<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-14" />

<title>Bayesian inference for a binomial proportion</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Bayesian inference for a binomial
proportion</h1>
<h4 class="author">Matthew Stephens</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">January 14, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree7cdbb3e1f42d125d5735f0c1b6058d871d75540atargetblank7cdbb3ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/7cdbb3e1f42d125d5735f0c1b6058d871d75540a" target="_blank">7cdbb3e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree7cdbb3e1f42d125d5735f0c1b6058d871d75540atargetblank7cdbb3ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/7cdbb3e1f42d125d5735f0c1b6058d871d75540a" target="_blank">7cdbb3e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/bayes_beta_binomial.Rmd</code>)
and HTML (<code>docs/bayes_beta_binomial.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/29fc90bf7b81d8e23b79ecd0ae3381f49197d01b/analysis/bayes_beta_binomial.Rmd" target="_blank">29fc90b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-14
</td>
<td>
Fixed use of propto for the likelihood.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d5f5166a2bfa55bb538a3bcea83ea8d05c27bcb0/analysis/bayes_beta_binomial.Rmd" target="_blank">d5f5166</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
Fixed a link.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/9f0f1a7b467a766dea1be91e01f4c672a2789947/docs/bayes_beta_binomial.html" target="_blank">9f0f1a7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
Ran wflow_publish("analysis/bayes_beta_binomial.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/98191c48d3cd08bfdeeecfe9a422b1cf38622737/analysis/bayes_beta_binomial.Rmd" target="_blank">98191c4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
A few more small updates to the bayes_beta_binomial vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/2853350626eb478d53c05072d70ed18c6e09cb50/analysis/bayes_beta_binomial.Rmd" target="_blank">2853350</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
A few minor updates for the bayes_beta_binomial vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/dfbb324352c1376b2ca6e5e0793570bccb889c6e/analysis/bayes_beta_binomial.Rmd" target="_blank">dfbb324</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
Updated links in bayes_beta_binomial vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/a2212405e4ac7f933f5917fae9375c3b98998391/docs/bayes_beta_binomial.html" target="_blank">a221240</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-09
</td>
<td>
Push a bunch of updates to the webpages.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/25e1cf5973a15cc8da373318d89aed09ee4b0607/analysis/bayes_beta_binomial.Rmd" target="_blank">25e1cf5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-08
</td>
<td>
Adding pdf versions of three other vignettes.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/5f62ee6cf86b77e9f45e0c6b6e3d6b97c7faf675/docs/bayes_beta_binomial.html" target="_blank">5f62ee6</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0cd28bd3e7aea2258f9c3a55e2e0060c55403123/analysis/bayes_beta_binomial.Rmd" target="_blank">0cd28bd</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
workflowr::wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/34bcc51fef570dd7d5a70859f262c983fed17381/docs/bayes_beta_binomial.html" target="_blank">34bcc51</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5fbc8b5c658203baf6765874b24794d41da622a8/analysis/bayes_beta_binomial.Rmd" target="_blank">5fbc8b5</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Update workflowr project with wflow_update (version 0.4.0).
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/8e616835a4eccac7c4e0a8eeec86fdcefb1525ab/docs/bayes_beta_binomial.html" target="_blank">8e61683</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-03-03
</td>
<td>
rendered html using wflow_build(all=TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d6741417d44f168473b77d41bba75ddf1acce30b/analysis/bayes_beta_binomial.Rmd" target="_blank">d674141</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-02-26
</td>
<td>
typos, refs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/aedeb9283ce291d458f6bc73b084acf4e3af0200/docs/bayes_beta_binomial.html" target="_blank">aedeb92</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/e7101d7af1e70cca8cd7e05ea04b8fce429a5c59/analysis/bayes_beta_binomial.Rmd" target="_blank">e7101d7</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-19
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/7bc1873ac28c9cf1bff1675c2228dbfc81bcac5a/docs/bayes_beta_binomial.html" target="_blank">7bc1873</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/35d9a16ee6254ac98636b16efc3b225a4479eb7c/analysis/bayes_beta_binomial.Rmd" target="_blank">35d9a16</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-28
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/5d88119b372a6262b70f6994335711d2904a21e7/docs/bayes_beta_binomial.html" target="_blank">5d88119</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/b48dd9c4b9d4117d2273df81bb8736c9ed10c44f/analysis/bayes_beta_binomial.Rmd" target="_blank">b48dd9c</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-25
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/bayes_beta_binomial.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This vignette illustrates how to perform Bayesian inference for a
continuous parameter, specifically a binomial proportion. Specifically,
it illustrates the mechanics of how we actually calculate the posterior
distribution.</p>
<p>You should be familiar with the concepts of the <a
href="https://pcarbo.github.io/fiveMinuteStats/likelihood_function.html">likelihood
function</a> and <a
href="https://pcarbo.github.io/fiveMinuteStats/bayes_multiclass.html">Bayesian
inference for discrete random variables</a>. You should also be familiar
with the binomial distribution and the <a
href="https://pcarbo.github.io/fiveMinuteStats/beta.html">Beta
distribution</a>.</p>
</div>
<div id="motivation" class="section level1">
<h1>Motivation</h1>
<p><strong>Technical note:</strong> To simplify this problem, I have
assumed that elephants are haploid, which they are not. If you do not
know what this means then you should simply ignore this comment.</p>
<p>Suppose we sample 100 elephants from a population, and measure their
DNA at one location (“locus”) in their genome, where there are two types
(“alleles”). We label these alleles as “0” and “1”.</p>
<p>In my sample, I observe that 30 of the elephants have the 1 allele,
and 70 have the 0 allele. What can I say about the frequency (<span
class="math inline">\(q\)</span>) of the 1 allele in the population?</p>
</div>
<div id="bayesian-inference-calculating-the-posterior"
class="section level1">
<h1>Bayesian inference: calculating the posterior</h1>
<p>Here we are doing inference for a parameter, <span
class="math inline">\(q\)</span>, that can, in principle, take any value
between 0 and 1. That is, we are doing inference for a “continuous”
parameter. Bayesian inference for a continuous parameter proceeds in
essentially the same way as Bayesian inference for a <a
href="https://pcarbo.github.io/fiveMinuteStats/bayes_multiclass.html">discrete
quantity</a> except that probability mass functions get replaced by
densities.</p>
<p>Remember Bayes Theorem: <span class="math display">\[
\text{posterior} \propto
\text{likelihood} \times \text{prior}.
\]</span> To apply this, we need to have both the prior distribution and
the likelihood.</p>
<div id="likelihood" class="section level2">
<h2>Likelihood</h2>
<p>Here, the likelihood for <span class="math inline">\(q\)</span> is
<span class="math display">\[
L(q) := p(D \mid q) = q^{30} (1-q)^{70},
\]</span> where <span class="math inline">\(D\)</span> here denotes the
data. This expression comes from the fact that the data consist of 30
“1” alleles (each of which occur with probability <span
class="math inline">\(q\)</span>) and 70 “0” alleles (each of which
occur with probability <span class="math inline">\(1-q\)</span>), and we
assume that the samples are independent. (You might have heard this
likelihood called the “binomial likelihood”, because it arises when the
data come from a binomial distribution.)</p>
</div>
<div id="prior" class="section level2">
<h2>Prior</h2>
<p>Recall that the prior distribution is a distribution that is supposed
to reflect what we know about <span class="math inline">\(q\)</span>
before (“prior to”) to seeing the data. For illustration, we will assume
a uniform prior on <span class="math inline">\(q\)</span>, <span
class="math display">\[
q \sim U[0,1].
\]</span> That is, <span class="math display">\[
p(q) = 1, \quad q \in [0,1].
\]</span></p>
<p>This prior says many things. For example, it says that, before seeing
the data, a <span class="math inline">\(q\)</span> less than 0.5 is just
as plausible as a <span class="math inline">\(q\)</span> greater than
0.5. It also says that a <span class="math inline">\(q\)</span> less
than 0.1 is just as plausible as a <span
class="math inline">\(q\)</span> greater than 0.9 or a <span
class="math inline">\(q\)</span> between 0.5 and 0.5. If for some reason
these are not equally plausible, then you should use a different prior.
However, in practice it is sometimes (but not always!) the case that the
results of Bayesian inference are robust to the choice of prior
distribution, and in such cases it is common not to worry too much about
minor discrepancies between what you believe and what the prior
implies.</p>
<p>For now, we are simply aiming to show how the Bayesian calculations
are done under this prior.</p>
</div>
<div id="posterior-calculation" class="section level2">
<h2>Posterior calculation</h2>
<p>Using Bayes Theorem to combine the prior distribution and the
likelihood, we obtain <span class="math display">\[
p(q \mid D) \propto p(D \mid q) \,
p(q) = q^{30} (1-q)^{70}, \quad q \in [0,1].
\]</span> Because <span class="math inline">\(q\)</span> is a continuous
parameter, this is called the <em>posterior density</em> for <span
class="math inline">\(q\)</span>.</p>
<p>Now the final trick is to notice that this density, <span
class="math inline">\(q^{30}
(1-q)^{70}\)</span> is <em>exactly the density of a <a
href="https://pcarbo.github.io/fiveMinuteStats/beta.html">Beta
distribution</a>.</em> Specifically, it is the density of a <span
class="math inline">\(\mathrm{Beta}(31,71)\)</span> distribution. So the
posterior distribution for <span class="math inline">\(q\)</span> is
<span class="math inline">\(\mathrm{Beta}(31,71)\)</span>, which we
write as <span class="math inline">\(q \mid D \sim
\mathrm{Beta}(31,71)\)</span>.</p>
<p>This kind of trick is common in Bayesian inference: you look at the
posterior density and “recognize” it as a distribution you know. It
turns out that the number of distributions in common use is relatively
small, so you only need to learn a few distributions to get sufficiently
good at this trick for practical purposes. For example, it is a good
start to be able to recognize the following distributions: exponential,
binomial, Poisson, Gamma, Beta, Dirichlet and normal. If your posterior
distribution does not look like one of these, then you may be in a
situation where you need to use computational methods like <a
href="https://pcarbo.github.io/fiveMinuteStats/Importance_sampling.html">Importance
Sampling</a> or <a
href="https://pcarbo.github.io/fiveMinuteStats/MH-examples1.html">Markov
chain Monte Carlo</a> to do your computations.</p>
<p>In this case, we were lucky: the posterior distribution is a
distribution that we recognize, and this means we can do many
calculations very easily. R has many built-in functions for calculations
with the Beta distribution, and many analytic properties have been
derived (e.g., <a
href="https://en.wikipedia.org/wiki/Beta_distribution">see the Wikipedia
page</a>.) And we can use this result to summarize and interpret the
posterior distribution, as we illustrate <a
href="https://pcarbo.github.io/fiveMinuteStats/summarize_interpret_posterior.html">here</a>.</p>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<ul>
<li><p>To compute the posterior density of a continuous parameter, up to
a normalizing constant, you multiply the likelihood by the prior
density.</p></li>
<li><p>In simple cases, you may find that the result is the density of a
distribution you recognize. If so, you can often use known properties of
that distribution to compute quantities of interest. See <a
href="https://pcarbo.github.io/fiveMinuteStats/summarize_interpret_posterior.html">here</a>
for an example.</p></li>
<li><p>In cases where you do not recognize the posterior distribution,
you may need to use computational methods like importance sampling or
Markov chain Monte Carlo to compute quantities of interest.</p></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
