<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-02-09" />

<title>Gibbs sampling for clustering genetic data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gibbs sampling for clustering genetic
data</h1>
<h4 class="author">Matthew Stephens and Peter Carbonetto</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">February 9, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstreeb4158aafed0389aa32c4129f15086bbf60c138bctargetblankb4158aaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/b4158aafed0389aa32c4129f15086bbf60c138bc" target="_blank">b4158aa</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstreeb4158aafed0389aa32c4129f15086bbf60c138bctargetblankb4158aaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/b4158aafed0389aa32c4129f15086bbf60c138bc" target="_blank">b4158aa</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/bernoulli_poisson_process_cache/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/gibbs_structure_simple.Rmd</code>) and HTML
(<code>docs/gibbs_structure_simple.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/b4158aafed0389aa32c4129f15086bbf60c138bc/analysis/gibbs_structure_simple.Rmd" target="_blank">b4158aa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
Updated code in gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0a3811c39e920dbd95166eecbc4002cfe6d04fac/analysis/gibbs_structure_simple.Rmd" target="_blank">0a3811c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
Added derivation for F | z, X to gibbs_structure_simple.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d9ca0f3c2f2ac5ddb7eac11fdd7a4692827f0d19/analysis/gibbs_structure_simple.Rmd" target="_blank">d9ca0f3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
Yet more improvements to gibbs_structure_simple.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c726ca73d16235ddfb6158aa2f9ed999580c7f1a/analysis/gibbs_structure_simple.Rmd" target="_blank">c726ca7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
Derived full conditional for z in gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/7f1ce9798e9e7708a273f8b067e624a6635c6836/analysis/gibbs_structure_simple.Rmd" target="_blank">7f1ce97</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
More improvements to the gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5fee37826d1119674caf1ba72327a25f44e29c9e/analysis/gibbs_structure_simple.Rmd" target="_blank">5fee378</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
A few more updates to the notation in gibbs_structure_simple.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/9bd8607463242c175efce6f29fd5e1002c021fd1/analysis/gibbs_structure_simple.Rmd" target="_blank">9bd8607</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-02-09
</td>
<td>
Updated some of notation in gibbs_structure_simple.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/64191f00070aa929796d8e39d5b9df6948d9381f/analysis/gibbs_structure_simple.Rmd" target="_blank">64191f0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-28
</td>
<td>
Fixed a comment in the gibbs_structure_simple code.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/0d912ff5e3f80d16bc1fb9f408ea8c8d3d4cd00a/docs/gibbs_structure_simple.html" target="_blank">0d912ff</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
Ran wflow_publish("analysis/gibbs_structure_simple.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/828954ad059508561db3f5f03e73ed795f8d5435/analysis/gibbs_structure_simple.Rmd" target="_blank">828954a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
wflow_publish("analysis/gibbs_structure_simple.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5db660cb44e0111940227aded255fd473b5fdb47/analysis/gibbs_structure_simple.Rmd" target="_blank">5db660c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
Further updates to the gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/897345239819d7e1cd82ec796368ddb7b29147c1/analysis/gibbs_structure_simple.Rmd" target="_blank">8973452</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
Yet more updates to the gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/db927dfbe7f726a9e8f278468e5a6acbaacef5c2/analysis/gibbs_structure_simple.Rmd" target="_blank">db927df</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
Fixed a couple links.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8bded2d53a3f4de234dce53e2fa08599dbbf5b6d/analysis/gibbs_structure_simple.Rmd" target="_blank">8bded2d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-24
</td>
<td>
More updates to the gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5314608087850d6823787edd6a2756d2ccd92bd9/analysis/gibbs_structure_simple.Rmd" target="_blank">5314608</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Updates to the gibbs_structure_simple vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/5f62ee6cf86b77e9f45e0c6b6e3d6b97c7faf675/docs/gibbs_structure_simple.html" target="_blank">5f62ee6</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0cd28bd3e7aea2258f9c3a55e2e0060c55403123/analysis/gibbs_structure_simple.Rmd" target="_blank">0cd28bd</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
workflowr::wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/fbb6c48b8689dd25ff8fddb896c82c8c76e2b451/docs/gibbs_structure_simple.html" target="_blank">fbb6c48</a>
</td>
<td>
stephens999
</td>
<td>
2018-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8183f19c3fe1f98c7ed2c1f7228097405b686610/analysis/gibbs_structure_simple.Rmd" target="_blank">8183f19</a>
</td>
<td>
stephens999
</td>
<td>
2018-05-02
</td>
<td>
workflowr::wflow_publish("analysis/gibbs_structure_simple.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/gibbs_structure_simple.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>Be familiar with <a
href="https://pcarbo.github.io/fiveMinuteStats/LR_and_BF.html">Bayesian
inference for the two class problem</a> and <a
href="https://pcarbo.github.io/fiveMinuteStats/bayes_conjugate.html">conjugate
Bayesian analysis for a binomial proportion</a>.</p>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>Suppose we observe genetic data on a sample of <span
class="math inline">\(n\)</span> elephants at <span
class="math inline">\(J\)</span> genetic markers (or “loci”) in the
genome. For simplicity, we will assume the elephants are haploid; that
is, they have just one copy of their genome. And we will assume that
there are just two genetic types (“alleles”) at each locus, which we
will label as 0 and 1.</p>
<p>We will further assume that there are two type of elephant: forest
elephants and savanna elephants, and that the allele frequencies in
forest elephants are different from those in savanna elephants, but that
the allele frequencies for each of these two groups are unknown. Also,
we do not know which samples are forest elephants and which are savanna
elehants. Our goal is to infer both these sets of unknowns: (i) which
elephants are forest and which are savanna; (ii) what are the allele
frequencies in each group of elephants.</p>
</div>
<div id="notation" class="section level1">
<h1>Notation</h1>
<p>Let <span class="math inline">\(x_i\)</span> denote the genetic data
for individual <span class="math inline">\(i\)</span> (<span
class="math inline">\(i = 1,\dots,
n\)</span>). Thus, <span class="math inline">\(x_i\)</span> is a binary
vector (a vector of zeros and ones) of length <span
class="math inline">\(J\)</span>. Let <span
class="math inline">\(X\)</span> denote the combined genetic data, a
<span class="math inline">\(n \times J\)</span> matrix.</p>
<p>Let <span class="math inline">\(z_i \in \{1, 2\}\)</span> denote the
group (forest vs. savanna) of individual <span
class="math inline">\(i\)</span>, and let <span
class="math inline">\(z\)</span> denote the vector <span
class="math inline">\(z = (z_1, \dots, z_n)\)</span>.</p>
<p>Let <span class="math inline">\(f_{jk}\)</span> denote the frequency
of the 1 allele at locus <span class="math inline">\(j\)</span> in group
<span class="math inline">\(k\)</span> (<span class="math inline">\(j =
1, \dots, J\)</span>, <span class="math inline">\(k = 1, 2\)</span>).
(Here, 1 = forest and 2 = savanna.) Let <span
class="math inline">\(f_k\)</span> denote the vector <span
class="math inline">\((f_{1k},
\dots, f_{Jk})\)</span>, and let <span class="math inline">\(F\)</span>
denote all the (unknown) allele frequencies <span
class="math inline">\(F = (f_1, f_2)\)</span>. (So <span
class="math inline">\(F\)</span> is a <span class="math inline">\(J
\times 2\)</span> matrix.)</p>
<p>With this notation in place, we can state the problem, which is to
infer the unknowns <span class="math inline">\(F\)</span> and <span
class="math inline">\(z\)</span> from the genetic data <span
class="math inline">\(X\)</span>.</p>
</div>
<div id="model" class="section level1">
<h1>Model</h1>
<p>To perform Bayesian inference for <span
class="math inline">\(z\)</span> and <span
class="math inline">\(F\)</span>, we need to specify the likelihood,
<span class="math inline">\(p(X \mid z, F)\)</span>, and a prior
distribution, <span class="math inline">\(p(z, F)\)</span>.</p>
<div id="likelihood" class="section level2">
<h2>Likelihood</h2>
<p>For each individual, we will assume that if we knew its group of
origin, and we knew the allele frequencies in each group, then the
genetic data at the different markers are independent draws from the
relevant allele frequencies. This is exactly the model assumed <a
href="https://pcarbo.github.io/fiveMinuteStats/likelihood_ratio_simple_models.html">here</a>.
In mathematical notation, we assume <span class="math display">\[
p(x_i \mid z_i, F) = p(x_i \mid z_i, f_{j1}, f_{j2}) =
\prod_{j=1}^J f_{j{z_i}}^{x_{ij}}(1 - f_{j{z_i}})^{1 - x_{ij}}.
\]</span> All the subscripts here make this difficult to read. To make
things easier to read, we can replace <span
class="math inline">\(z_i\)</span> with <span
class="math inline">\(k\)</span>: <span class="math display">\[
p(x_i \mid z_i = k, F) = p(x_i \mid z_i, f_{j1}, f_{j2}) =
\prod_{j=1}^J f_{jk}^{x_{ij}}(1 - f_{jk})^{1 - x_{ij}}.
\]</span></p>
<p>We will further assume that the different individuals (elephants) are
independent: <span class="math display">\[
p(X \mid z, F) = \prod_{i=1}^n p(x_i \mid z_i, F).
\]</span></p>
<p>This completes specification of the likelihood.</p>
</div>
<div id="prior" class="section level2">
<h2>Prior</h2>
<p>We will assume that <span class="math inline">\(F\)</span> and <span
class="math inline">\(z\)</span> are <em>a priori</em> independent, so
<span class="math inline">\(p(F, z)
= p(F) \, p(z)\)</span>. This assumption seems reasonable: before seeing
the genetic data (<span class="math inline">\(X\)</span>), telling you
the allele frequencies in the two groups would not tell you anything
about the group membership of the elephants. (Of course, after seeing
the genetic data <span class="math inline">\(X\)</span>, the allele
frequencies would help classify the individuals, so <span
class="math inline">\(F\)</span> and <span
class="math inline">\(z\)</span> are not going to be <em>a
posteriori</em> independent. However, here we are concerned with the
prior, not the posterior.)</p>
<p>For the prior on <span class="math inline">\(F\)</span>, we will
further assume that the allele frequencies in each group at each locus
are independent, so <span class="math display">\[
p(F) = \prod_{k=1}^2 \prod_{j=1}^J p(f_{jk}).
\]</span> This assumption could be improved, but at the cost of
considerable extra complexity, and so we stick with independence for
now. Also, for simplicity we will assume a uniform prior distribution
for <span class="math inline">\(f_{jk}\)</span>, so <span
class="math inline">\(p(f_{jk}) = 1\)</span>.</p>
<p>For <span class="math inline">\(z\)</span>, we will assume that the
origin of each individual is independent, and that the probability of
arising from group <span class="math inline">\(k\)</span> is <span
class="math inline">\(q_k\)</span>. So <span class="math display">\[
p(z) = \prod_{i=1}^n p(z_i),
\]</span> and <span class="math inline">\(p(z_i = k) = q_k\)</span>, in
which <span class="math inline">\(q_1\)</span>, <span
class="math inline">\(q_2\)</span> are taken to be known (that is, we
will not estimate them from the data). Again, this assumption could be
improved, but we start here for simplicity.</p>
</div>
</div>
<div id="computation" class="section level1">
<h1>Computation</h1>
<p>Our goal is to compute (or sample from) the posterior distribution
<span class="math inline">\(p(z, F \mid X)\)</span>, which by Bayes
Theorem is <span class="math display">\[
p(z, F \mid X) \propto p(X \mid z, F) \, p(z, F).
\]</span></p>
<p>One way to sample from this distribution is to implement a Gibbs
sampler for <span class="math inline">\(z, F\)</span>. This requires us
to be able to do two things:</p>
<ol style="list-style-type: decimal">
<li><p>Sample from <span class="math inline">\(p(z \mid F,
X)\)</span>.</p></li>
<li><p>Sample from <span class="math inline">\(p(F \mid z,
X)\)</span>.</p></li>
</ol>
<p>These are called the “full conditional distributions”, or “full
conditions”, for <span class="math inline">\(z\)</span> and <span
class="math inline">\(F\)</span>, respectively. The use of the word
“full” here indicates that they are conditional on <em>everything
else</em> (the data and all the other parameters).</p>
<div id="another-way-to-describe-this-gibbs-sampler"
class="section level2">
<h2>Another way to describe this Gibbs sampler</h2>
<p>Because of the conditional independence assumptions we made above,
the Gibbs sampler we just described also can be described in the
following way:</p>
<ol style="list-style-type: decimal">
<li><p>For <span class="math inline">\(i = 1, \ldots, n\)</span>, sample
from <span class="math inline">\(p(z_i \mid F, X)\)</span>.</p></li>
<li><p>For <span class="math inline">\(j = 1, \ldots, J\)</span> and
<span class="math inline">\(k = 1, 2\)</span>, sample from <span
class="math inline">\(p(f_{jk} \mid z, X)\)</span>.</p></li>
</ol>
<p>This algorithm is equivalent to the algorithm above.</p>
<p>In other words, because of the conditional independence assumptions
we made, it turns out that sampling <span
class="math inline">\(z\)</span> given <span class="math inline">\((F,
X)\)</span> is the same as sampling the group <span
class="math inline">\(z_i\)</span> for a single elephant, one at a time.
Similarly, because of the conditional independence assumptions we made,
it turns out that sampling <span class="math inline">\(F\)</span> given
<span class="math inline">\((z, X)\)</span> is the same as sampling an
allele frequency <span class="math inline">\(f_{jk}\)</span> for a
single locus and a single group one at a time. This makes the Gibbs
sampler particularly easy to implement.</p>
<p>We will verify these two statements next.</p>
</div>
<div id="full-conditional-for-z" class="section level2">
<h2>Full conditional for z</h2>
<p>From the modeling assumptions made above, we know that <span
class="math display">\[
\begin{aligned}
p(z \mid F, X) &amp;\propto p(X \mid z, F) \, p(z) \, p(F) \\
&amp;= \prod_{i=1}^n \{ p(x_i \mid z_i, F) \, p(z_i) \} \, p(F).
\end{aligned}
\]</span></p>
<p>So we see that the full conditional for <span class="math inline">\(z
= (z_1, \dots, z_n)\)</span> factorizes over <span
class="math inline">\(i\)</span> into terms that depend only on <span
class="math inline">\(z_i\)</span> and not the other elements of <span
class="math inline">\(z\)</span>; that is, <span class="math display">\[
p(z \mid F, X) \propto \prod_{i=1}^n \phi_i(z_i),
\]</span> in which the functions <span
class="math inline">\(\phi_i(z_i)\)</span> are <span
class="math display">\[
\phi_i(z_i) = p(x_i \mid z_i, F) \, p(z_i).
\]</span></p>
<p>This implies that the <span class="math inline">\(z_i\)</span> are
<em>conditionally independent</em> given <span class="math inline">\(X,
F\)</span>, which is very convenient as it means we can compute their
conditional distribution just by computing the marginals: <span
class="math display">\[
p(z_i = k \mid F, X) \propto p(x_i \mid z_i = k, F) \, p(z_i = k).
\]</span> The exact expression for <span class="math inline">\(p(z_i = k
\mid F, X)\)</span> is derived below.</p>
</div>
<div id="full-conditional-for-f" class="section level2">
<h2>Full conditional for F</h2>
<p>From the modeling assumptions made above, we know that <span
class="math display">\[
p(F \mid z, X) \propto p(X \mid z, F) \, p(F) \, p(z) =
p(z) \,
\prod_{j=1}^J \prod_{k=1}^2 p(f_{jk}) \times \big\{
{\textstyle \prod_{i=1}^n p(x_{ij} \mid z_i, f_{j1}, f_{j2})} \big\}.
\]</span> So we see that the full conditional for <span
class="math inline">\(F\)</span> factorizes over <span
class="math inline">\(j\)</span> and <span
class="math inline">\(k\)</span> into terms that depend only on <span
class="math inline">\(f_{jk}\)</span> and not any of the other allele
frequencies. In other words, <span class="math display">\[
p(F \mid z, X) \propto \prod_{j=1}^J \prod_{k=1}^2
\psi_{jk}(f_{jk}),
\]</span> in which the functions <span
class="math inline">\(\psi_{jk}(f_{jk})\)</span> are <span
class="math display">\[
\psi_{jk}(f_{jk}) = p(f_{jk})
\prod_{i=1}^n p(x_{ij} \mid z_i, f_{j1}, f_{j2}).
\]</span> The exact expression for <span class="math inline">\(p(f_{jk}
\mid z, X)\)</span> is derived below.</p>
</div>
</div>
<div id="simulate-some-data" class="section level1">
<h1>Simulate some data</h1>
<p>To illustrate, let’s simulate data from this model:</p>
<pre class="r"><code>set.seed(33)

# Simulate n samples from the &quot;simple haploid&quot; model.
r_simplemix &lt;- function (n, F, q1) {
  J &lt;- nrow(F)
  z &lt;- sample(2,prob = c(q1,1-q1),size = n,replace = TRUE)
  X &lt;- matrix(0,n,J)
  for (i in 1:n) {
    k &lt;- z[i]
    X[i,] &lt;- rbinom(J,rep(1,J),F[,k])
  }
  rownames(X) &lt;- paste0(&quot;i&quot;,1:n)
  colnames(X) &lt;- paste0(&quot;j&quot;,1:J)
  names(z) &lt;- paste0(&quot;i&quot;,1:n)
  return(list(X = X,z = z))
}

# Simulate a data set with n = 50 elephants and J = 6 genetic markers.
q1 &lt;- 0.5
F_true &lt;- cbind(c(0.500,0.500,0.500,0.500,0.500,0.500),
                c(0.001,0.999,0.001,0.999,0.001,0.999))
rownames(F_true) &lt;- paste0(&quot;j&quot;,1:6)
colnames(F_true) &lt;- c(&quot;k1&quot;,&quot;k2&quot;)
sim &lt;- r_simplemix(50,F_true,q1)
X &lt;- sim$X
z_true &lt;- sim$z</code></pre>
</div>
<div id="gibbs-sampler-code" class="section level1">
<h1>Gibbs sampler code</h1>
<p>This function implements the main loop:</p>
<pre class="r"><code># Inputs are: X, an n x J binary matrix of n alleles at J loci; q1,
# the prior probability of z = 1; and T, the length of the Markov
# chain to simulate. The output is the T states of the Markov chain.
run_gibbs_sampler &lt;- function (X, q1, T = 100) {
  n &lt;- nrow(X)
  J &lt;- ncol(X)
  states &lt;- list(z = matrix(0,n,T),
                 F = array(0,c(J,2,T)))
  z &lt;- sample(2,n,replace = TRUE)
  for (t in 1:T) {
    F &lt;- sample_F_given_z(X,z)
    z &lt;- sample_z_given_F(X,F,q1)
    states$z[,t] &lt;- z
    states$F[,,t] &lt;- F
  }
  return(states)
}</code></pre>
<p>This is the function for sampling <span
class="math inline">\(F\)</span> given <span class="math inline">\(z,
X\)</span>:</p>
<pre class="r"><code># Draws the allele frequencies F randomly from the posterior
# distribution p(F | X, z). The output is a J x 2 matrix, where J is
# the number of markers.
sample_F_given_z &lt;- function (X, z) {
  J &lt;- ncol(X)
  F &lt;- matrix(0,J,2)
  for (k in 1:2) {
    i &lt;- which(z == k)
    n &lt;- length(i)
    if (n == 0)
      n1 &lt;- 0
    else
      n1 &lt;- colSums(X[i,])
    n0 &lt;- n - n1
    F[,k] &lt;- rbeta(J,n1 + 1,n0 + 1)
  }
  return(F)
}</code></pre>
<p>This is the function for sampling <span
class="math inline">\(z\)</span> given <span class="math inline">\(F,
X\)</span>:</p>
<pre class="r"><code># Draws the elephant group assignments z randomly from the posterior
# distribution p(z | X, F). The output is a vector of length n, where
# n is the number of elephants.
sample_z_given_F &lt;- function (X, F, q1) {
  n &lt;- nrow(X)
  z &lt;- rep(0,n)
  for (i in 1:n) {
    logw &lt;- compute_loglik_times_prior_z_given_F(X[i,],F,q1)
    a    &lt;- max(logw)
    post &lt;- exp(logw - a)
    post &lt;- normalize(post)
    z[i] &lt;- sample(2,1,prob = post)
  }
  return(z)
}

# This function is used in sample_z_given_F.
normalize &lt;- function (x)
  x/sum(x)

# Another function used in sample_z_given_F.
compute_loglik_times_prior_z_given_F &lt;- function (x, F, q1) {
  f1 &lt;- F[,1]
  f2 &lt;- F[,2]
  q2 &lt;- 1 - q1
  return(c(log(q1) + sum((1-x)*log(1-f1)) + sum(x*log(f1)),
           log(q2) + sum((1-x)*log(1-f2)) + sum(x*log(f2))))
}</code></pre>
<p>Try the Gibbs sampler on the data simulated above:</p>
<pre class="r"><code>q1 &lt;- 0.5
T &lt;- 100
states &lt;- run_gibbs_sampler(X,q1,T)</code></pre>
<p>Plot the state of the Markov chain over time:</p>
<pre class="r"><code>par(mfrow = c(1,2))
J &lt;- ncol(X)
plot(1:T,states$F[1,1,],col = &quot;red&quot;,type = &quot;l&quot;,xlab = &quot;time&quot;,
     ylab = &quot;allele frequency&quot;,ylim = c(0,1))
for (j in 1:J) {
  lines(1:T,states$F[j,1,],col = &quot;red&quot;)
  lines(1:T,states$F[j,2,],col = &quot;blue&quot;)
}
image(t(states$z),col = c(&quot;red&quot;,&quot;blue&quot;),xlab = &quot;time&quot;,ylab = &quot;elephant&quot;)</code></pre>
<p><img src="figure/gibbs_structure_simple.Rmd/plot-markov-chain-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Compare the true groupings to the posterior mean of <span
class="math inline">\(z\)</span>:</p>
<pre class="r"><code>head(cbind(true = z_true,est = rowMeans(states$z)),n = 8)
#    true  est
# i1    2 1.01
# i2    2 1.03
# i3    2 1.03
# i4    1 1.98
# i5    1 1.98
# i6    1 1.96
# i7    2 1.02
# i8    2 1.03</code></pre>
<p>Compare the true allele frequencies to the posterior mean of <span
class="math inline">\(F\)</span>.</p>
<pre class="r"><code>cbind(true = F_true[,1],est = rowMeans(states$F[,1,]))
#    true        est
# j1  0.5 0.06798217
# j2  0.5 0.94418090
# j3  0.5 0.07689037
# j4  0.5 0.91541484
# j5  0.5 0.06153185
# j6  0.5 0.94577366
cbind(true = F_true[,2],est = rowMeans(states$F[,2,]))
#     true       est
# j1 0.001 0.4378161
# j2 0.999 0.4638687
# j3 0.001 0.6680624
# j4 0.999 0.3569845
# j5 0.001 0.3802636
# j6 0.999 0.4621341</code></pre>
<p>The estimates look wrong, but actually this is an example of the
<em>label switching problem.</em> Switching the labels—that is,
replacing <span class="math inline">\(k = 1\)</span> with 2 and
replacing <span class="math inline">\(k = 2\)</span> with 1—should fix
the problem and make the estimates align better to the true values:</p>
<pre class="r"><code>z_switch &lt;- 2 - states$z
head(cbind(true = z_true,est = rowMeans(z_switch)),n = 8)
#    true  est
# i1    2 0.99
# i2    2 0.97
# i3    2 0.97
# i4    1 0.02
# i5    1 0.02
# i6    1 0.04
# i7    2 0.98
# i8    2 0.97
cbind(true = F_true[,1],est = rowMeans(states$F[,2,]))
#    true       est
# j1  0.5 0.4378161
# j2  0.5 0.4638687
# j3  0.5 0.6680624
# j4  0.5 0.3569845
# j5  0.5 0.3802636
# j6  0.5 0.4621341
cbind(true = F_true[,2],est = rowMeans(states$F[,1,]))
#     true        est
# j1 0.001 0.06798217
# j2 0.999 0.94418090
# j3 0.001 0.07689037
# j4 0.999 0.91541484
# j5 0.001 0.06153185
# j6 0.999 0.94577366</code></pre>
</div>
<div id="additional-derivations" class="section level1">
<h1>Additional derivations</h1>
<div id="full-conditional-for-z-1" class="section level2">
<h2>Full conditional for z</h2>
<p>Given that the full conditional for <span
class="math inline">\(z\)</span> factorizes over the individual
elephants <span class="math inline">\(i\)</span>, without loss of
generality we can derive the full conditional assuming a data set with a
single elephant. Dropping the unneeded the <span
class="math inline">\(i\)</span> subscripts, this is <span
class="math display">\[
p(z = k \mid F, x) \propto p(x \mid z = k, F) \, p(z = k).
\]</span> Plugging in the expression for the likelihood and prior above,
this is <span class="math display">\[
p(z = k \mid F, x) \propto
q_k \times \prod_{j=1}^J f_{jk}^{x_j}(1-f_{jk})^{1-x_j}.
\]</span> Note that an additional normalization step is needed to
actually obtain probabilities.</p>
</div>
<div id="full-conditional-for-f-1" class="section level2">
<h2>Full conditional for F</h2>
<p>Given that the full conditional for <span
class="math inline">\(F\)</span> factorizes over the individual loci
<span class="math inline">\(j\)</span>, without loss of generality we
can derive the full conditional assumine a data set with a single locus
or marker, so <span class="math inline">\(F\)</span> is just two allele
frequencies, <span class="math inline">\(f_1\)</span> and <span
class="math inline">\(f_2\)</span>. We further found out that we can
focus on a single <span class="math inline">\(k\)</span> since the full
conditional also factorized over <span class="math inline">\(k\)</span>.
Dropping the unneeded <span class="math inline">\(j\)</span> subscripts,
this is <span class="math display">\[
p(f_k \mid z, x) \propto p(f_k) \prod_{i=1}^n p(x_i \mid z_i, f).
\]</span> Since we assumed a uniform prior, this simplifies to <span
class="math display">\[
p(f_k \mid z, x) \propto \prod_{i=1}^n p(x_i \mid z_i, f).
\]</span></p>
<p>Next, a useful observation is that <span class="math inline">\(p(x_i
\mid z_i, f)\)</span> only involves <span
class="math inline">\(f_k\)</span> if <span class="math inline">\(z_i =
k\)</span>. So we only need to actually consider the likelihood terms
for which the elephant <span class="math inline">\(i\)</span> is
assigned to group <span class="math inline">\(k\)</span>. Denoting <span
class="math inline">\(n_{0k}\)</span> as the number of elephants in
group <span class="math inline">\(k\)</span> with the 0 allele, and
denoting <span class="math inline">\(n_{1k}\)</span> as the number of
elephants in group <span class="math inline">\(k\)</span> with the 1
allele, this can be written more simply as <span class="math display">\[
p(f_k \mid z, x) \propto f_k^{n_{1k}} (1 - f_k)^{n_{0k}}.
\]</span> Recall, from our previous derivations, this means that <span
class="math display">\[
f_k \mid z, x \sim \mathrm{Beta}(n_{1k} + 1, n_{0k} + 1).
\]</span></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
