<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-06" />

<title>Comparing two models with a likelihood ratio</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparing two models with a likelihood
ratio</h1>
<h4 class="author">Matthew Stephens</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">January 6, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree2d6d41e3a5ce2be246b9962df54e2ad3d605b56ftargetblank2d6d41ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/2d6d41e3a5ce2be246b9962df54e2ad3d605b56f" target="_blank">2d6d41e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree2d6d41e3a5ce2be246b9962df54e2ad3d605b56ftargetblank2d6d41ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/2d6d41e3a5ce2be246b9962df54e2ad3d605b56f" target="_blank">2d6d41e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/#bayes_multiclass.Rmd#

Unstaged changes:
    Modified:   analysis/likelihood_ratio_simple_continuous_data.Rmd
    Modified:   analysis/likelihood_ratio_simple_models.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/likelihood_ratio_simple_models.Rmd</code>) and HTML
(<code>docs/likelihood_ratio_simple_models.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/a361e11f18b8997cb1db99688946aed83e6370e9/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">a361e11</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
A bunch of small edits to the bayes_multiclass vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/9781cd816a271a3cff79a634d6bd677f70bf8f36/docs/likelihood_ratio_simple_models.html" target="_blank">9781cd8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
Ran wflow_publish("analysis/likelihood_ratio_simple_models.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0593f14b39283c56b8f9c3a83c4802ebc2c0ebe8/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">0593f14</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
wflow_publish("analysis/likelihood_ratio_simple_models.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/fc8b1d8a4e3be1def37036773f4dc03cfce48803/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">fc8b1d8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
Some edits to the exercises in the likelihood_ratio_simple_models
vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8a0af869caeaf7c30e378744ba4a7cc26ad361b5/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">8a0af86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
A few more small edits to the likelihood_ratio_simple_models vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c67ccfc8b567412dad382fef0a2f73861e3dcda7/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">c67ccfc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
A few more edits to the likelihood_ratio_simple_models vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/306bf92a4941ccf950d7d9d4f2363e2d6d354e83/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">306bf92</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
A few minor edits to the likelihood_ratio_simple_models vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/7a8ffe4a6d2f6e537f3b90bbee2ca2ee362bd167/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">7a8ffe4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Created pdf version of LR_and_BF vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/7a8ffe4a6d2f6e537f3b90bbee2ca2ee362bd167/docs/likelihood_ratio_simple_models.html" target="_blank">7a8ffe4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Created pdf version of LR_and_BF vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/49b0c3374b189e362b0ed6c253ce9bd4c318d715/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">49b0c33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Added a link.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/49b0c3374b189e362b0ed6c253ce9bd4c318d715/docs/likelihood_ratio_simple_models.html" target="_blank">49b0c33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Added a link.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8c88ffc1f9b9f0dde6683ca4b39b655f5b4aaa59/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">8c88ffc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
More basic infrastructure work.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/8c88ffc1f9b9f0dde6683ca4b39b655f5b4aaa59/docs/likelihood_ratio_simple_models.html" target="_blank">8c88ffc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
More basic infrastructure work.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/599a8d7cc512f20373c23308b393204f36d8a6e4/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">599a8d7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
A few small adjustments to the workflowr settings.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/04994f5dd7af60de1d5764dc2142d8f0b6e5ed23/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">04994f5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Added basic Makefile infrastructure.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/6cefaaa585fd4d9bcdd0417395c225afb3db4a54/docs/likelihood_ratio_simple_models.html" target="_blank">6cefaaa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c8270f5b0c31cd1d6af15fc9c8ff707e5e3a735e/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">c8270f5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
workflowr::wflow_publish("analysis/likelihood_ratio_simple_models.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/35b85cdb459273820cbe3927d4ce486d68730085/docs/likelihood_ratio_simple_models.html" target="_blank">35b85cd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/96cc4f4220c378dd3a244d3041c61b4eee434047/docs/likelihood_ratio_simple_models.html" target="_blank">96cc4f4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/cd79db895de543985d0a10388c9ab53382824205/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">cd79db8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
wflow_publish("analysis/likelihood_ratio_simple_models.Rmd",
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c345a1a8050d37146ff43f44cb721058a1424b2e/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">c345a1a</a>
</td>
<td>
Nan Xiao
</td>
<td>
2024-03-03
</td>
<td>
Fix broken and moved URL
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/5f62ee6cf86b77e9f45e0c6b6e3d6b97c7faf675/docs/likelihood_ratio_simple_models.html" target="_blank">5f62ee6</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0cd28bd3e7aea2258f9c3a55e2e0060c55403123/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">0cd28bd</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
workflowr::wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/219e6c48e6523064117ac7ec73e2813823e3fcfc/docs/likelihood_ratio_simple_models.html" target="_blank">219e6c4</a>
</td>
<td>
stephens999
</td>
<td>
2018-03-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/34bcc51fef570dd7d5a70859f262c983fed17381/docs/likelihood_ratio_simple_models.html" target="_blank">34bcc51</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/2aa70fe2e48f71cb1c6e59773c50246a7deff083/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">2aa70fe</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Manually udpate some Rmd files.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5fbc8b5c658203baf6765874b24794d41da622a8/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">5fbc8b5</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Update workflowr project with wflow_update (version 0.4.0).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/391ba3c0cf210ea8de8125ee8063557ceff55080/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">391ba3c</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Remove front and end matter of non-standard templates.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/8e616835a4eccac7c4e0a8eeec86fdcefb1525ab/docs/likelihood_ratio_simple_models.html" target="_blank">8e61683</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-03-03
</td>
<td>
rendered html using wflow_build(all=TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d6741417d44f168473b77d41bba75ddf1acce30b/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">d674141</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-02-26
</td>
<td>
typos, refs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/f7af26933583caac8ce2a126c0ed34ebadfb4e48/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">f7af269</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-02-25
</td>
<td>
knitr style citations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/3be128c87db057b34a05c5487946df966bac6914/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">3be128c</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-02-24
</td>
<td>
start bibtex ref, urls etc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/c3b365a48f7fc76cf83ade425916d31e547d8077/docs/likelihood_ratio_simple_models.html" target="_blank">c3b365a</a>
</td>
<td>
John Blischak
</td>
<td>
2017-01-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/67a857539cd8a4029e899208aed19da6c78b4b36/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">67a8575</a>
</td>
<td>
John Blischak
</td>
<td>
2017-01-02
</td>
<td>
Use external chunk to set knitr chunk options.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5ec12c73e37fc3c7ec04a31f6d85eede0f6c436e/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">5ec12c7</a>
</td>
<td>
John Blischak
</td>
<td>
2017-01-02
</td>
<td>
Use session-info chunk.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/6a2fec7ffd24ba5decfc1a0ee96a3707458e319b/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">6a2fec7</a>
</td>
<td>
stephens999
</td>
<td>
2016-04-07
</td>
<td>
fix typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/b61683013b70cbb8e9e0124f4a6648676a1d8a4c/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">b616830</a>
</td>
<td>
Arjun Biddanda
</td>
<td>
2016-01-18
</td>
<td>
Fixed Typo between fF in table and in code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d6f4bea8fc4f77f1c125eecc341aa60bafcbdd8e/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">d6f4bea</a>
</td>
<td>
stephens999
</td>
<td>
2016-01-12
</td>
<td>
add examples
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/6c7d0e0a1ba46fa42ce678adf929d3a5eda792bd/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">6c7d0e0</a>
</td>
<td>
stephens999
</td>
<td>
2016-01-08
</td>
<td>
updated exercises
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/93476efa7fe12ec6613b1efd7f7e2aa96e7ddffb/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">93476ef</a>
</td>
<td>
stephens999
</td>
<td>
2016-01-07
</td>
<td>
update LR to make clear it is for discrete data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8f6123275fb86d72c80124c97e8c5ed6f09ad07b/analysis/likelihood_ratio_simple_models.Rmd" target="_blank">8f61232</a>
</td>
<td>
stephens999
</td>
<td>
2016-01-07
</td>
<td>
add simple LR example
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/likelihood_ratio_simple_models.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>This document introduces the concept of likelihood ratio for
comparing two simple (“fully specified”) statistical models, one of the
most fundamental concepts in statistical inference.</p>
</div>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>Be familiar with basic concepts from probability, particularly the
concept of a probability distribution.</p>
</div>
<div id="motivating-problem" class="section level1">
<h1>Motivating Problem</h1>
<p><strong>Technical note:</strong> To simplify this problem I have
assumed that elephants are [haploid][plody], which they are not. If you
do not know what this means you should simply ignore this comment.</p>
<p>There are two subspecies of African Elephant: savannah and forest
elephants, which differ in their genetic makeup. Interpol have seized an
illegally-smuggled elephant tusk, and they want to know which subspecies
of elephant the tusk came from. To try to answer this, they collect DNA
from the tusk and measure it at a number of locations (“markers” in
genetics jargon) along the elephant genome. At each marker, the DNA can
be one of two types (“alleles” in genetics jargon), which for simplicity
we will label 0 and 1. So the available data on the tusk from one
elephant might look something like this:</p>
<table>
<thead>
<tr class="header">
<th align="center">marker</th>
<th align="center">allele</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">1</td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="center">2</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">3</td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="center">4</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">5</td>
<td align="center">0</td>
</tr>
<tr class="even">
<td align="center">6</td>
<td align="center">1</td>
</tr>
</tbody>
</table>
<p>Interpol also have information on the frequency of each allele in
each of the two subspecies — this was obtained by measuring the DNA of a
large number of savanna elephants and a large number of forest
elephants. We will use <span class="math inline">\(f_{Sj}\)</span> and
<span class="math inline">\(f_{Fj}\)</span> to denote the frequency of
the “1” allele at marker <span class="math inline">\(j\)</span> in
savanna and forest elephants, respectively. (And since there are only
two alleles, the frequency of the “0” allele is <span
class="math inline">\(1 - f_{Sj}\)</span> and <span
class="math inline">\(1 - f_{Fj}\)</span>, respectively.) Here is a
table of this information:</p>
<table>
<thead>
<tr class="header">
<th align="center">marker, <span class="math inline">\(j\)</span></th>
<th align="center"><span class="math inline">\(f_{Sj}\)</span></th>
<th align="center"><span class="math inline">\(f_{Fj}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">1</td>
<td align="center">0.40</td>
<td align="center">0.80</td>
</tr>
<tr class="even">
<td align="center">2</td>
<td align="center">0.12</td>
<td align="center">0.20</td>
</tr>
<tr class="odd">
<td align="center">3</td>
<td align="center">0.21</td>
<td align="center">0.11</td>
</tr>
<tr class="even">
<td align="center">4</td>
<td align="center">0.12</td>
<td align="center">0.17</td>
</tr>
<tr class="odd">
<td align="center">5</td>
<td align="center">0.02</td>
<td align="center">0.23</td>
</tr>
<tr class="even">
<td align="center">6</td>
<td align="center">0.32</td>
<td align="center">0.25</td>
</tr>
</tbody>
</table>
<p>The question before us is: <em>which subspecies of elephant did the
tusk come from, and how confident should we be in this
conclusion?</em></p>
<p>To get some intuition, let us examine the data at the first few
markers. At marker 1, our tusk has the allele 1. This allele is less
common in savanna elephants than forest elephants (40% of savanna
elephants carry this allele, vs. 80% of forest elephants), so this
observation seems to support the sample coming from forest. However, at
the same time 40% of savanna samples carry this allele, so it remains
plausible that the sample came from savanna.</p>
<p>Moving to marker 2, our tusk has the allele 0, which is more common
in savanna (88%) than forest elephants (80%). And at marker 3 the tusk
has allele 1 which is also more common in savanna (21% vs. 11%). So, in
contrast to marker 1, the data at these two markers are more consistent
with the tusk coming from a savanna elephant than a forest elephant.</p>
<p>We could continue in this way, but the point should be clear: each
marker contains some information, but not a huge amount, and sometimes
the information points in different directions. By taking a statistical
approach to this problem, we aim to quantify this kind of information,
and to efficiently combine the information across markers to come to an
overall conclusion.</p>
<p>(This is a simplified version of a real problem: Interpol and other
authorities want to know the origin of poached tusks to help focus
efforts on curbing this illegal activity. In practice, they are
interested in much finer-level discrimination, and measure many genetic
markers to get more information. See <a
href="https://doi.org/10.1073/pnas.0609714104">Wasser et al (2007)</a>
and <a href="https://doi.org/10.1111/j.1523-1739.2008.01012.x">Wasser et
al (2008)</a> for more details.)</p>
</div>
<div id="solution" class="section level1">
<h1>Solution</h1>
<p>We can phrase this problem as a “model comparison” problem. We have
data <span class="math inline">\(X=x\)</span> from our tusk, and we have
two different models for how those data might have arisen: it could have
been sampled from a savanna elephant, or it could have been sampled from
a forest elephant. We will use <span class="math inline">\(M_S\)</span>
and <span class="math inline">\(M_F\)</span> as shorthand for these two
models. A key point is that these two models imply different probability
distributions for <span class="math inline">\(X\)</span>: some values of
<span class="math inline">\(X\)</span> are more common under <span
class="math inline">\(M_S\)</span> and others are more common under
<span class="math inline">\(M_F\)</span>.</p>
<p>Denoting the probability mass functions of these two distributions as
<span class="math inline">\(p(x \mid M_S)\)</span> and <span
class="math inline">\(p(x \mid M_F)\)</span>, and assuming the data at
different markers are independent, these probability distributions are
<span class="math display">\[
p(x \mid M_S) = \prod_{j=1}^J f_{Sj}^{x_j} (1-f_{Sj})^{1-x_j}
\]</span> and <span class="math display">\[p(x \mid M_F) = \prod_{j=1}^J
f_{Fj}^{x_j} (1-f_{Fj})^{1-x_j},
\]</span> where <span class="math inline">\(J = 6\)</span> is the number
of markers, and the values of <span
class="math inline">\(f_{Sj}\)</span> and <span
class="math inline">\(f_{Fj}\)</span> are given in the table above.</p>
<p>Note that a key feature of these models is that they are <em>fully
specified.</em> This means there are no unknown values in the
probability distributions. Comparing fully-specified models is the
simplest kind of model comparison, and so is a good place to start.</p>
<div id="the-likelihood-ratio" class="section level2">
<h2>The likelihood ratio</h2>
<p>A key idea is that a useful summary of how strongly the data <span
class="math inline">\(x\)</span> support one model vs. another model is
given by the “likelihood ratio” (LR).</p>
<p>The LR comparing two fully-specified models is simply the ratio of
the probabilities of the data under each model. (In saying the
“probability of the data”, we are assuming that the data and models
involved are discrete. For continuous data and models, the LR is the
ratio of the <em>probability densities</em> of the two models evaluated
at the data, as discussed in detail <a
href="https://pcarbo.github.io/fiveMinuteStats/likelihood_ratio_simple_continuous_data.html">here</a>.)</p>
<p>The name “likelihood ratio” comes from the fact that the probability
of the data under each model is called the “likelihood” for that model.
I recommended saying “likelihood for” the model, or “likelihood under”
the model, rather than “likelihood of” the model, to help avoid
confusing likelihood with probability.</p>
<p>That is, given data <span class="math inline">\(x\)</span>, the
likelihood for a fully-specified (discrete) model <span
class="math inline">\(M\)</span> is defined as <span
class="math display">\[
L(M) := p(x \mid M),
\]</span> where <span class="math inline">\(p(x \mid M)\)</span> denotes
the probability mass function for model <span
class="math inline">\(M\)</span>. And the likelihood ratio comparing two
fully-specified (discrete) models <span
class="math inline">\(M_1\)</span> vs. <span
class="math inline">\(M_0\)</span> is defined as <span
class="math display">\[
\mathrm{LR}(M_1, M_0) := \frac{L(M_1)}{L(M_0)}.
\]</span> Note that the likelihood for model <span
class="math inline">\(M\)</span> depends on the data <span
class="math inline">\(x\)</span>. To make this dependence explicit, the
likelihood is sometimes written as <span
class="math inline">\(L(M;x)\)</span> instead of just <span
class="math inline">\(L(M)\)</span>.</p>
<p>Large values of <span class="math inline">\(\mathrm{LR}(M_1,
M_0)\)</span> indicate that the data are much more probable under model
<span class="math inline">\(M_1\)</span> than under model <span
class="math inline">\(M_0\)</span>, and so indicate support for <span
class="math inline">\(M_1\)</span>. Conversely, small values of <span
class="math inline">\(\mathrm{LR}(M_1, M_0)\)</span> indicate support
for model <span class="math inline">\(M_0\)</span>.</p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<p>Using the numbers in the tables above, we can compute the likelihood
and LR for <span class="math inline">\(M_S\)</span> vs. <span
class="math inline">\(M_F\)</span>:</p>
<pre class="r"><code>x &lt;- c(1,0,1,0,0,1)
fS &lt;- c(0.40,0.12,0.21,0.12,0.02,0.32)
fF &lt;- c(0.8,0.2,0.11,0.17,0.23,0.25)
L &lt;- function (f, x)
  prod(f^x * (1-f)^(1-x))
LR &lt;- L(fS,x)/L(fF,x)
LR
# [1] 1.81359</code></pre>
<p>So <span class="math inline">\(\mathrm{LR}(M_S, M_F)\)</span> is
1.8135904. This means that the data favor the tusk coming from a savanna
elephant by a factor of about 1.8. This is a fairly modest factor — not
large enough to draw a convincing conclusion. We will have more to say
about interpreting LRs, and what values might be considered “convincing”
later.</p>
<p>Note that we have deliberately focused on the likelihood ratio, and
not the actual likelihood values themselves. This is because actual
likelihood values are generally not useful — it is only the ratios that
matter when comparing the models. One way of thinking about this is that
the actual likelihood values are very context dependent, and so
likelihoods from different data sets are not comparable with one
another. However, the meaning of the likelihood ratio is in some sense
consistent across contexts: LR = 1.8 means that the data favour the
first model by a factor of 1.8, whatever the context.</p>
</div>
</div>
<div id="the-inverse-likelihood-ratio-and-the-log-likelihood-ratio"
class="section level1">
<h1>The inverse likelihood ratio and the log-likelihood ratio</h1>
<p>Notice that, from the definition of the LR, the LR for <span
class="math inline">\(M_0\)</span> vs. <span
class="math inline">\(M_1\)</span> is the just inverse of the LR for
<span class="math inline">\(M_1\)</span> vs. <span
class="math inline">\(M_0\)</span>. That is, <span
class="math display">\[
\mathrm{LR}(M_0,M_1) = \frac{1}{\mathrm{LR}(M_1,M_0)}.
\]</span> So for example, if <span
class="math inline">\(\mathrm{LR}(M_1,M_0) = 0.01\)</span>, then <span
class="math inline">\(\mathrm{LR}(M_0,M_1) = 100\)</span>, and the data
favour <span class="math inline">\(M_0\)</span> vs. <span
class="math inline">\(M_1\)</span> by a factor of 100.</p>
<p>For many reasons, it is common to work with the log-likelihood ratio,
<span class="math inline">\(\mathrm{LLR} := \log\mathrm{LR}\)</span>.
Usually mathematicians work with logarithms in base <span
class="math inline">\(e\)</span>, and we will use that convention unless
otherwise stated. So for example, if LR = 1.8, then <span
class="math inline">\(\mathrm{LLR} = \log 1.8 =\)</span> 0.5877867.</p>
<p>Although the usual convention is to use base <span
class="math inline">\(e\)</span>, it can sometimes be useful to work
with logarithms in base 10 to make the inverse logarithm operation
easier for human calculation. For example, if I tell you that the LLR,
base 10, is 3, then you can immediately tell me that the LR is
1,000.</p>
</div>
<div id="exercises" class="section level1">
<h1>Exercises</h1>
<ol style="list-style-type: decimal">
<li><p>You are playing a game with a friend. The friend has two
six-sided dice, one blue and one green. The sides on the blue die are
numbered 1, 2, 3, 3, 3 and 4. The sides on the green die are labeled 1,
2, 2, 3, 4 and 4. He picks one of the dice without telling you, and
rolls it 10 times, obtaining the results 3, 3, 2, 3, 1, 2, 3, 3, 4, 3.
Looking at these results, does intuition say that they support the green
die or the blue die? Strongly or weakly? Phrase the problem as a model
comparison problem. State your modelling assumptions, and compute a
likelihood ratio. Does it support your intuition?</p></li>
<li><p>(Read the whole question before starting!)</p>
<ol style="list-style-type: lower-alpha">
<li><p>Perform the following simulation study. Simulate 1,000 tusks
(values of <span class="math inline">\(x\)</span>) from each of the
models <span class="math inline">\(M_S\)</span> and <span
class="math inline">\(M_F\)</span>. For each simulated tusk, compute the
LR for <span class="math inline">\(M_S\)</span> vs. <span
class="math inline">\(M_F\)</span>, so you have computed 2,000 LRs. Now
consider using the LR to classify each tusk as being from a savanna or a
forest elephant. Recall that large values for LR indicate support for
<span class="math inline">\(M_S\)</span>, so a natural classification
rule is “classify as savanna if <span class="math inline">\(\mathrm{LR}
&gt; c\)</span>, otherwise classify as forest”, for some threshold <span
class="math inline">\(c\)</span>. Plot the misclassification rate
(number of tusks wrongly classified divided by 2,000) for this rule, as
<span class="math inline">\(c\)</span> ranges from 0.01 to 100. What
value of <span class="math inline">\(c\)</span> minimizes the
misclassification rate? [Hint: the plot will look best if you do things
on the log scale, so you could let <span
class="math inline">\(\log_{10}c\)</span> vary from <span
class="math inline">\(-2\)</span> to <span
class="math inline">\(2\)</span> using an equally spaced grid, then plot
the misclassification rate on the Y axis vs. <span
class="math inline">\(\log_{10}c\)</span> on the X axis.]</p></li>
<li><p>Repeat the above simulation study using 100 tusks from <span
class="math inline">\(M_S\)</span> and 1,900 tusks from <span
class="math inline">\(M_F\)</span>. What value of <span
class="math inline">\(c\)</span> minimizes the misclassification rate?
Comment on this result.</p></li>
</ol></li>
</ol>
<p><em>Note:</em> it is good practice when writing computer code for a
simulation study like this to separate the code into small functions
each of which accomplishes a clearly defined task. So for example, you
should write a function to simulate data, and another to compute the LR
(possibly using the “L” function defined above), another to compute the
misclassification rate, <em>etc.</em> It is also good practice to
comment your functions, e.g., start by describing the inputs and
outputs. I strongly suggest using <a
href="https://r-pkgs.org/man.html">roxygen2</a> format. See <a
href="https://kbroman.org/pkg_primer/pages/docs.html">Karl Broman’s
tutorial</a> for example.</p>
<p>DO NOT REPEAT THE SIMULATION STUDY BY COPYING AND PASTING CODE AND
MODIFYING A FEW NUMBERS. GET INTO THE HABIT OF DOING IT PROPERLY.</p>
<p>To be more explicit, at the top level your code should include a
function that looks something like this:</p>
<pre class="r"><code>#&#39; @title Run Elephant Tusk Simulation STudy
#&#39;
#&#39; @description Simulate data from forest and savanna tusks with
#&#39;   prespecified frequencies, then use LRs to classify them.
#&#39;
#&#39; @param nS Number of savanna tusks.
#&#39;
#&#39; @param nF Number of forest tusks.
#&#39;
#&#39; @param fS Allele frequencies of the savanna elephants.
#&#39;
#&#39; @param fF Allele frequencies of the forest elephants.
#&#39;
#&#39; @param lc Vector of `log(c)` values of the LR cutoff to use.
#&#39;
#&#39; @param seed Set the random seed using `set.seed` for reproducibility.
#&#39;
#&#39; @return A plot of misclassification rate vs. lc.
#&#39;
#&#39; @example
#&#39; simulation_study(1000,1000,
#&#39;                  c(0.40,0.12,0.21,0.12,0.02,0.32),
#&#39;                  c(0.8,0.2,0.11,0.17,0.23,0.25),
#&#39;                  seq(-2,2,length.out = 100),seed)
#&#39;
elephant_simulation_study &lt;- function (nS, nF, fS, fF, lc, seed) {
  # Add your code here.
}</code></pre>
<p>Then calling this function twice should provide the results needed to
complete 2a and 2b.</p>
<ol start="3" style="list-style-type: decimal">
<li>Now consider modifying the model to allow for errors in the data.
Specifically, suppose that there is an error probability of 0.02 when
measuring each marker: with probability 0.98 you observe the true <span
class="math inline">\(x_j\)</span>, but with probability 0.02 an error
is made, and you observe <span class="math inline">\(1-x_j\)</span>.
Assume that errors occur independently at each marker <span
class="math inline">\(j\)</span>. Let <span
class="math inline">\(M&#39;_S\)</span> and <span
class="math inline">\(M&#39;_F\)</span> denote the models with this
error process incorporated. Derive expressions for the likelihood for
<span class="math inline">\(M&#39;_S\)</span> and <span
class="math inline">\(M&#39;_F\)</span>, and compute the LR for the tusk
data given at the beginning of the example.</li>
</ol>
</div>
<div id="summary-1" class="section level1">
<h1>Summary</h1>
<p>Repeat after me:</p>
<ol style="list-style-type: decimal">
<li><p>“The likelihood for a model is the probability of the data under
the model.”</p></li>
<li><p>“Individual likelihood values are mostly irrelevant; it is
likelihood ratios that matter.”</p></li>
<li><p>“If the likelihood ratio for model 1 vs. model 2 is <span
class="math inline">\(x &gt; 1\)</span>, this means the data favour
model 1 by a factor of <span class="math inline">\(x\)</span>.” (Or, if
<span class="math inline">\(x &lt; 1\)</span>, this means the data
favour model 2 by a factor of <span
class="math inline">\(1/x\)</span>.)</p></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
