<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-23" />

<title>Simple examples of the Metropolis-Hastings algorithm</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simple examples of the Metropolis-Hastings
algorithm</h1>
<h4 class="author">Matthew Stephens</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">January 23, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree7007f5b12926ac52bfb2ab50be1cb42b899791aetargetblank7007f5ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/7007f5b12926ac52bfb2ab50be1cb42b899791ae" target="_blank">7007f5b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree7007f5b12926ac52bfb2ab50be1cb42b899791aetargetblank7007f5ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/7007f5b12926ac52bfb2ab50be1cb42b899791ae" target="_blank">7007f5b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/MH-examples1.Rmd</code>) and HTML
(<code>docs/MH-examples1.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/7007f5b12926ac52bfb2ab50be1cb42b899791ae/analysis/MH-examples1.Rmd" target="_blank">7007f5b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
wflow_publish("analysis/MH-examples1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/db49bb68b58c8765d2fdce66776ed3a7187a96e7/docs/MH-examples1.html" target="_blank">db49bb6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Ran wflow_publish("analysis/MH-examples1.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/3e3246e2d5c52478216745545068a28bbcc03721/analysis/MH-examples1.Rmd" target="_blank">3e3246e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Updates to the rest of MH-examples1 vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c88c3e6dbc0ada19a54bc9bea1d01bedc3927327/analysis/MH-examples1.Rmd" target="_blank">c88c3e6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Updates to Example 2 from the MH-examples1 vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/ddbf10ab9f80f25956491da2a5d32e56f5ef405b/analysis/MH-examples1.Rmd" target="_blank">ddbf10a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Updated to example 1 in MH-examples1 vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/8fa865852b0acd619408cee55e00c075aa295578/analysis/MH-examples1.Rmd" target="_blank">8fa8658</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
<td>
Added rule for MH-examples1 vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/60908f53248452ecad277c755cf4becbbd3fca44/docs/MH-examples1.html" target="_blank">60908f5</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2023-07-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/366ef4e72c3776f6ce4a2740943439d42f40cba6/analysis/MH-examples1.Rmd" target="_blank">366ef4e</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2023-07-18
</td>
<td>
workflowr::wflow_publish("analysis/MH-examples1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/c6b20e2e0e8d3583049e3679dac281b7d9648982/docs/MH-examples1.html" target="_blank">c6b20e2</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/09da4d463240dcabbcb60ee6449e45d00b29436e/analysis/MH-examples1.Rmd" target="_blank">09da4d4</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2022-04-26
</td>
<td>
workflowr::wflow_publish("MH-examples1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/154369251207824c4985b873cefbb5ab8c572fbb/docs/MH-examples1.html" target="_blank">1543692</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2022-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/dadc5496a705dbff4fa8ff7a69b65a9682fed41c/analysis/MH-examples1.Rmd" target="_blank">dadc549</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2022-04-26
</td>
<td>
workflowr::wflow_publish("MH-examples1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/5f62ee6cf86b77e9f45e0c6b6e3d6b97c7faf675/docs/MH-examples1.html" target="_blank">5f62ee6</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0cd28bd3e7aea2258f9c3a55e2e0060c55403123/analysis/MH-examples1.Rmd" target="_blank">0cd28bd</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2019-03-31
</td>
<td>
workflowr::wflow_publish(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/34bcc51fef570dd7d5a70859f262c983fed17381/docs/MH-examples1.html" target="_blank">34bcc51</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c7339fc202993b0a10b36ce87c55024a37166435/analysis/MH-examples1.Rmd" target="_blank">c7339fc</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Minor updates.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/5fbc8b5c658203baf6765874b24794d41da622a8/analysis/MH-examples1.Rmd" target="_blank">5fbc8b5</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Update workflowr project with wflow_update (version 0.4.0).
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/391ba3c0cf210ea8de8125ee8063557ceff55080/analysis/MH-examples1.Rmd" target="_blank">391ba3c</a>
</td>
<td>
John Blischak
</td>
<td>
2017-03-06
</td>
<td>
Remove front and end matter of non-standard templates.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/8e616835a4eccac7c4e0a8eeec86fdcefb1525ab/docs/MH-examples1.html" target="_blank">8e61683</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-03-03
</td>
<td>
rendered html using wflow_build(all=TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/d6741417d44f168473b77d41bba75ddf1acce30b/analysis/MH-examples1.Rmd" target="_blank">d674141</a>
</td>
<td>
Marcus Davy
</td>
<td>
2017-02-26
</td>
<td>
typos, refs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/a0309bfe160a35733aa6fa3ed695663c138669a1/docs/MH-examples1.html" target="_blank">a0309bf</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0a96ec4f599f56f6968bec3929589cf0212b40a9/analysis/MH-examples1.Rmd" target="_blank">0a96ec4</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-16
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/d54f3267b777968c40b42fd9351db3833038523c/docs/MH-examples1.html" target="_blank">d54f326</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/0207794438b4d4032490b687c7bff809f58fca13/analysis/MH-examples1.Rmd" target="_blank">0207794</a>
</td>
<td>
stephens999
</td>
<td>
2017-02-15
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/ad91ca8f203c74cd4fcad4fbc993dc454d9b4904/docs/MH-examples1.html" target="_blank">ad91ca8</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-25
</td>
<td>
update html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/19974203a28306b2d710ef8d21941d225d98b394/analysis/MH-examples1.Rmd" target="_blank">1997420</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-25
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/ea1f50ed0b1dacf540c3abc0ee55a2b0b882c973/docs/MH-examples1.html" target="_blank">ea1f50e</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-25
</td>
<td>
update MH examples
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/ba83f29db669bd3a222afe795735a023c5332596/analysis/MH-examples1.Rmd" target="_blank">ba83f29</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-24
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/86ebaedfa2ba5f51f01513b6531ad701b31f8ebb/docs/MH-examples1.html" target="_blank">86ebaed</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/17331b4e1c393fe211544493940633603971ea93/analysis/MH-examples1.Rmd" target="_blank">17331b4</a>
</td>
<td>
stephens999
</td>
<td>
2017-01-24
</td>
<td>
Files commited by wflow_commit.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/MH-examples1.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>You should be familiar with the Metropolis-Hastings algorithm,
introduced <a
href="https://pcarbo.github.io/fiveMinuteStats/MH_intro.html">here</a>,
and elaborated on <a
href="https://pcarbo.github.io/fiveMinuteStats/MH_intro_02.html">here</a>.</p>
<div id="caveat-on-code" class="section level2">
<h2>Caveat on code</h2>
<p>Note that the code is designed to be readable by a beginner, rather
than “efficient”. The idea is that you can use this code to learn about
the basics of MCMC, but not as a model for how to program well in R!</p>
</div>
</div>
<div id="example-1-sampling-from-an-exponential-distribution-using-mcmc"
class="section level1">
<h1>Example 1: Sampling from an exponential distribution using MCMC</h1>
<p>Any MCMC scheme aims to produce (dependent) samples from a “target”
distribution. In this case we are going to use the exponential
distribution with mean 1 as our target distribution. Here we define this
function (on the log scale):</p>
<pre class="r"><code>log_exp_target &lt;- function (x)
  dexp(x,rate = 1,log = TRUE)</code></pre>
<p>The following code implements a simple M-H algorithm. (Note that the
parameter “log_target” is a function which computes the log of the
target distribution; you may be unfamiliar with the idea of passing a
function as a parameter, but it works just like any other type of
parameter)</p>
<pre class="r"><code>easyMCMC &lt;- function (log_target, niter, startval, proposalsd) {
  x &lt;- rep(0,niter)
  x[1] &lt;- startval
  for (i in 2:niter) {
    currentx &lt;- x[i-1]
    proposedx &lt;- rnorm(1,currentx,proposalsd) 
    A &lt;- exp(log_target(proposedx) - log_target(currentx))
    u &lt;- runif(1)
    if(u &lt; A)
      x[i] &lt;- proposedx
    else
      x[i] &lt;- currentx
  }
  return(x)
}</code></pre>
<p>Now we run the MCMC three times from different starting points and
compare the results:</p>
<pre class="r"><code>par(mfrow = c(1,2))
z1 &lt;- easyMCMC(log_exp_target,1000,3,1)
z2 &lt;- easyMCMC(log_exp_target,1000,1,1)
z3 &lt;- easyMCMC(log_exp_target,1000,5,1)
plot(z1,type = &quot;l&quot;,xlab = &quot;time&quot;,ylab = &quot;x&quot;)
lines(z2,col = 2)
lines(z3,col = 3)
plot(log_exp_target(z1),xlab = &quot;time&quot;,ylab = &quot;log target&quot;)
lines(log_exp_target(z2),col = 2)
lines(log_exp_target(z3),col = 3)</code></pre>
<p><img src="figure/MH-examples1.Rmd/run-easy-mcmc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-run-easy-mcmc-1">
Past versions of run-easy-mcmc-1.png
</button>
</p>
<div id="fig-run-easy-mcmc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/db49bb68b58c8765d2fdce66776ed3a7187a96e7/docs/figure/MH-examples1.Rmd/run-easy-mcmc-1.png" target="_blank">db49bb6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>par(mfrow = c(1,3))
maxz &lt;- max(c(z1,z2,z3))
hist(z1,breaks = seq(0,maxz,length.out = 20),xlab = &quot;x&quot;)
hist(z2,breaks = seq(0,maxz,length.out = 20),xlab = &quot;x&quot;)
hist(z3,breaks = seq(0,maxz,length.out = 20),xlab = &quot;x&quot;)</code></pre>
<p><img src="figure/MH-examples1.Rmd/summarize-easy-mcmc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-summarize-easy-mcmc-1">
Past versions of summarize-easy-mcmc-1.png
</button>
</p>
<div id="fig-summarize-easy-mcmc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/db49bb68b58c8765d2fdce66776ed3a7187a96e7/docs/figure/MH-examples1.Rmd/summarize-easy-mcmc-1.png" target="_blank">db49bb6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="exercise" class="section level2">
<h2>Exercise</h2>
<p>Use the function “easyMCMC” to explore the following:</p>
<ol style="list-style-type: lower-alpha">
<li><p>How do different starting values affect the MCMC scheme? (Try
some extreme starting points.)</p></li>
<li><p>What is the effect of having a bigger (or smaller) proposal
standard deviation? (Again, try some extreme values.)</p></li>
<li><p>Try changing the (log) target function to “log_target_bimodal”
below. What does this target distribution look like? What happens if the
proposal standard deviation is too small here? (e.g., try 1 and 0.1 and
see what happens)</p></li>
</ol>
<pre class="r"><code>log_target_bimodal &lt;- function (x)
  log(0.8*dnorm(x,-4,1) + 0.2*dnorm(x,4,1))</code></pre>
</div>
</div>
<div id="example-2-estimating-an-allele-frequency"
class="section level1">
<h1>Example 2: Estimating an allele frequency</h1>
<p>A standard assumption when modelling genotypes of biallelic loci
(e.g., loci with alleles <span class="math inline">\(A\)</span> and
<span class="math inline">\(a\)</span>) is that the population is
“mating at random”. From this assumption, it follows that the population
will be in “Hardy-Weinberg equilibrium” (HWE), which means that if <span
class="math inline">\(p\)</span> is the frequency of the <span
class="math inline">\(A\)</span> allele, then the genotypes <span
class="math inline">\(AA\)</span>, <span
class="math inline">\(Aa\)</span> and <span
class="math inline">\(aa\)</span> have frequencies <span
class="math inline">\(p^2\)</span>, <span
class="math inline">\(2p(1-p)\)</span> and <span
class="math inline">\((1-p)^2\)</span>, respectively.</p>
<p>A simple prior for <span class="math inline">\(p\)</span> is to
assume it is uniform on <span class="math inline">\([0,1]\)</span>.
Suppose that we sample <span class="math inline">\(n\)</span>
individuals, and observe <span class="math inline">\(n_{AA}\)</span>
individuals with genotype <span class="math inline">\(AA\)</span>, <span
class="math inline">\(n_{Aa}\)</span> individuals with genotype <span
class="math inline">\(Aa\)</span>, and <span
class="math inline">\(n_{aa}\)</span> individuals with genotype <span
class="math inline">\(aa\)</span>.</p>
<p>The following R code implements a M-H algorithm to sample from the
posterior distribution of <span class="math inline">\(p\)</span>. Try to
go through the code to see how it works.</p>
<p>The first function returns the (log) prior density:</p>
<pre class="r"><code>log_prior &lt;- function (p) {
  if((p &lt; 0) || (p &gt; 1))
    return(-Inf)
  else
    return(0)
}</code></pre>
<p>The second function returns the (log) likelihood:</p>
<pre class="r"><code>log_likelihood_hwe &lt;- function (p, nAA, nAa, naa)
  return(2*nAA*log(p) +
         nAa*log(2*p*(1-p)) +
         2*naa*log(1-p))</code></pre>
<p>And now this function implements the M-H algorithm:</p>
<pre class="r"><code>psampler &lt;- function (nAA, nAa, naa, niter, pstartval, pproposalsd) {
  p &lt;- rep(0,niter)
  p[1] &lt;- pstartval
  for (i in 2:niter) {
    currentp &lt;- p[i-1]
    newp &lt;- currentp + rnorm(1,0,pproposalsd)
    A &lt;- exp(log_prior(newp)
             - log_prior(currentp)
             + log_likelihood_hwe(newp,nAA,nAa,naa)
             - log_likelihood_hwe(currentp,nAA,nAa,naa))
    if (runif(1) &lt; A)
      p[i] &lt;- newp
    else
      p[i] &lt;- currentp
  }
  return(p)
}</code></pre>
<p>Try running this algorithm for <span class="math inline">\(n_{AA} =
50\)</span>, <span class="math inline">\(n_{Aa} = 21\)</span>, <span
class="math inline">\(n_{aa}
= 29\)</span>:</p>
<pre class="r"><code>z &lt;- psampler(50,21,29,10000,0.5,0.01)</code></pre>
<p>Now here’s some R code to compare the samples generated by the M-H
algorithm with the theoretical posterior. In this case, the theoretical
posterior is available analytically; since we observed <span
class="math inline">\(A\)</span> 121 times and <span
class="math inline">\(a\)</span> 79 times, the posterior for <span
class="math inline">\(p\)</span> is <span
class="math inline">\(\mathrm{Beta}(121 + 1,79 + 1)\)</span>.</p>
<pre class="r"><code>x &lt;- seq(0,1,length.out = 1000)
hist(z,breaks = 32,probability = TRUE,xlab = &quot;x&quot;,main = &quot;&quot;)
lines(x,dbeta(x,122,80))</code></pre>
<p><img src="figure/MH-examples1.Rmd/psampler-hist-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-psampler-hist-1">
Past versions of psampler-hist-1.png
</button>
</p>
<div id="fig-psampler-hist-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/db49bb68b58c8765d2fdce66776ed3a7187a96e7/docs/figure/MH-examples1.Rmd/psampler-hist-1.png" target="_blank">db49bb6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>You might also like to discard the first 5,000 samples as
“burnin”:</p>
<pre class="r"><code>hist(z[5001:10000],breaks = 16,probability = TRUE,xlab = &quot;x&quot;,main = &quot;&quot;)
lines(x,dbeta(x,122,80))</code></pre>
<p><img src="figure/MH-examples1.Rmd/unnamed-chunk-1-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/db49bb68b58c8765d2fdce66776ed3a7187a96e7/docs/figure/MH-examples1.Rmd/unnamed-chunk-1-1.png" target="_blank">db49bb6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="exercise-1" class="section level2">
<h2>Exercise</h2>
<p>Investigate how the starting point and proposal standard deviation
affect the convergence of the algorithm.</p>
</div>
</div>
<div
id="example-3-estimating-an-allele-frequency-and-inbreeding-coefficient"
class="section level1">
<h1>Example 3: Estimating an allele frequency and inbreeding
coefficient</h1>
<p>A slightly more complex alternative than HWE is to assume that there
is a tendency for people to mate with others who are slightly more
related than “random” (as might happen in a geographically structured
population, for example). This will result in an excess of homozygotes
compared with HWE. A simple way to capture this is to introduce an extra
parameter, the “inbreeding coefficient” <span
class="math inline">\(f\)</span>, and assume that the genotypes <span
class="math inline">\(AA\)</span>, <span
class="math inline">\(Aa\)</span> and <span
class="math inline">\(aa\)</span> have frequencies <span
class="math inline">\(fp + (1-f)p^2\)</span>, <span
class="math inline">\(2(1-f)
p(1-p)\)</span>, and <span class="math inline">\(f(1-p) +
(1-f)(1-p)^2\)</span>, respectively.</p>
<p>In most cases, it would be natural to treat <span
class="math inline">\(f\)</span> as a feature of the population, and
therefore assume <span class="math inline">\(f\)</span> is constant
across loci. For simplicity, we will consider just a single locus.</p>
<p>Note that both <span class="math inline">\(f\)</span> and <span
class="math inline">\(p\)</span> are constrained to lie between 0 and 1
(inclusive). A simple prior for each of these two parameters is to
assume that they are independent and uniform on <span
class="math inline">\([0,1]\)</span>.</p>
<p>Suppose that we sample <span class="math inline">\(n\)</span>
individuals, and observe <span class="math inline">\(n_{AA}\)</span>
individuals with genotype <span class="math inline">\(AA\)</span>, <span
class="math inline">\(n_{Aa}\)</span> individuals with genotype <span
class="math inline">\(Aa\)</span>, and <span
class="math inline">\(n_{aa}\)</span> individuals with genotype <span
class="math inline">\(aa\)</span>.</p>
<div id="exercise-2" class="section level2">
<h2>Exercise</h2>
<ol style="list-style-type: decimal">
<li>Write an M-H algorithm to sample from the joint distribution of
<span class="math inline">\(f\)</span> and <span
class="math inline">\(p\)</span>. <em>Hint:</em> Below is some code to
get you started. (You’ll need to fill in the “…”) As a first step, I
suggest writing the function to compute the log-likelihood for the
inbreeding model.</li>
</ol>
<pre class="r"><code>log_likelihood_inbreeding &lt;- function (...) {
  # ...
}</code></pre>
<pre class="r"><code>fpsampler &lt;- function (nAA, nAa, naa, niter, fstartval, pstartval,
                       fproposalsd, pproposalsd) {
  f &lt;- rep(0,niter)
  p &lt;- rep(0,niter)
  f[1] &lt;- fstartval
  p[1] &lt;- pstartval
  for (i in 2:niter) {
    currentf &lt;- f[i-1]
    currentp &lt;- p[i-1]
    # newf &lt;- currentf + ...
    # newp &lt;- currentp + ...
    # ...
  }
  return(list(f = f,p = p))
}</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Use the output of your M-H algorithm to compute point estimates for
<span class="math inline">\(f\)</span> and <span
class="math inline">\(p\)</span> (e.g., posterior means) as well as
interval estimates for <span class="math inline">\(f\)</span> and <span
class="math inline">\(p\)</span> (e.g., 90% posterior credible
intervals) when the data are <span class="math inline">\(n_{AA} =
50\)</span>, <span class="math inline">\(n_{Aa} = 21\)</span>, <span
class="math inline">\(n_{aa} = 29\)</span>.</li>
</ol>
</div>
</div>
<div id="addendum-gibbs-sampling" class="section level1">
<h1>Addendum: Gibbs sampling</h1>
<p>You could also tackle this problem with a Gibbs sampler (see also the
<a
href="https://pcarbo.github.io/fiveMinuteStats/gibbs1.html">gibbs1</a>
and <a
href="https://pcarbo.github.io/fiveMinuteStats/gibbs2.html">gibbs2</a>
vignettes). To do so, you will want to use the following “latent
variable” representation of the model: <span class="math display">\[
\begin{aligned}
\Pr(g_i = AA \mid z_i = 0) &amp;= p^2 \\
\Pr(g_i = Aa \mid z_i = 0) &amp;= 2p(1-p) \\
\Pr(g_i = aa \mid z_i = 0) &amp;= (1-p)^2,
\end{aligned}
\]</span> <span class="math display">\[
\begin{aligned}
\Pr(g_i = AA \mid z_i = 1) &amp;= p \\
\Pr(g_i = Aa \mid z_i = 1) &amp;= 0 \\
\Pr(g_i = aa \mid z_i = 1) &amp;= 1-p,
\end{aligned}
\]</span> and <span class="math display">\[
z_i \sim \mathrm{Bernoulli}(f).
\]</span></p>
<p>Note that summing over <span class="math inline">\(z_i\)</span> gives
the same model as above, e.g., <span class="math inline">\(\Pr(g_i = AA)
= fp + (1-f)p^2\)</span>. (As an exercise, show that the probabilities
of <span class="math inline">\(g_i = aa\)</span> and <span
class="math inline">\(g_i = Aa\)</span> from this “latent variable
augmented” model are the same as above.)</p>
<p><em>Reflection:</em> What does the latent variable <span
class="math inline">\(z_i\)</span> represent?</p>
<div id="exercise-3" class="section level2">
<h2>Exercise</h2>
<p>Using the latent variable representation above, implement a Gibbs
sampler to generate samples from the joint distribution of <span
class="math inline">\(z, f\)</span> and <span
class="math inline">\(p\)</span> given the genotype data <span
class="math inline">\(g\)</span>. <em>Hint:</em> this involves iterating
the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Sample <span class="math inline">\(z\)</span> from <span
class="math inline">\(p(z \mid g, f, p)\)</span>.</p></li>
<li><p>Sample <span class="math inline">\(f, p\)</span> from <span
class="math inline">\(p(f, p \mid g, z)\)</span>.</p></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
