<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-12" />

<title>Mixture models</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mixture models</h1>
<h4 class="author">Matthew Stephens</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">January 12, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree289127545edde17b45e7d89524c55b39e810e607targetblank2891275a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/289127545edde17b45e7d89524c55b39e810e607" target="_blank">2891275</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree289127545edde17b45e7d89524c55b39e810e607targetblank2891275a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/289127545edde17b45e7d89524c55b39e810e607" target="_blank">2891275</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/em_algorithm_01.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mixture_models_01.Rmd</code>) and
HTML (<code>docs/mixture_models_01.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/289127545edde17b45e7d89524c55b39e810e607/docs/mixture_models_01.html" target="_blank">2891275</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
Ran wflow_publish("analysis/mixture_models_01.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/620c8c6227fdda63f07ef89fbe8ffa9f5a5eacc7/analysis/mixture_models_01.Rmd" target="_blank">620c8c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
A few more minor updates to the mixture models vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/610d9e7f121d736785672914fd401219c8c59a0c/analysis/mixture_models_01.Rmd" target="_blank">610d9e7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
<td>
Updates to mixture_models_01 vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/a2212405e4ac7f933f5917fae9375c3b98998391/docs/mixture_models_01.html" target="_blank">a221240</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-09
</td>
<td>
Push a bunch of updates to the webpages.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/4e442397dc3a23b48687580de7c5b9929cd277d4/analysis/mixture_models_01.Rmd" target="_blank">4e44239</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-09
</td>
<td>
Added pdf versions of mixture_models_01 and em_algorithm_01 vignettes.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/803554afc74e2bbf64ed3fb4fba95df1c7cd64fc/docs/mixture_models_01.html" target="_blank">803554a</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/cc566d37c19e69fb2107063824c3beda60a0b620/analysis/mixture_models_01.Rmd" target="_blank">cc566d3</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-04-20
</td>
<td>
workflowr::wflow_publish("analysis/mixture_models_01.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/4ced11ac5653610fb2624cb3f021c5e777fc02c8/docs/mixture_models_01.html" target="_blank">4ced11a</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-01-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/4b1238c9497c8460cbb65667f667262b41a7af69/analysis/mixture_models_01.Rmd" target="_blank">4b1238c</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-01-26
</td>
<td>
workflowr::wflow_publish("mixture_models_01.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/f4bada48aaf826f2ba7b71bb1ffa0cf2f5bca092/docs/mixture_models_01.html" target="_blank">f4bada4</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-01-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/be59a00e50405a82727c31e1faefc52515626932/analysis/mixture_models_01.Rmd" target="_blank">be59a00</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-01-25
</td>
<td>
workflowr::wflow_publish("mixture_models_01.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/mixture_models_01.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<p>You should be familiar with basic probability, including the law of
total probability, the notion of distibution and density, and with
standard distributions (particularly the Gamma distribution).</p>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This vignette introduces the idea of a mixture model. These models
are widely used in statistics to model data where observations come from
a “mixture” of two or more different distributions. This vignette
introduces the basic idea of a mixture, its density/mass function, and
terminology such as <em>mixture proportions</em>, <em>mixture
components</em> and <em>latent variable representation.</em></p>
</div>
<div id="example" class="section level1">
<h1>Example</h1>
<p>We begin this vignette with an example. A medical screening test for
a disease involves measuring the concentration (<span
class="math inline">\(X\)</span>) of a protein in the blood. In normal
individuals, <span class="math inline">\(X\)</span> has a Gamma
distribution with mean 1 and shape 2 (so the scale parameter is 0.5, as
scale = mean/shape). In diseased individuals, the protein becomes
elevated, and <span class="math inline">\(X\)</span> has a Gamma
distribution with mean 2 and shape 2 (so scale = 1).</p>
<p>Suppose that in a particular population, 70% of individuals are
normal and 30% are diseased. What will be the overall distribution of
the protein levels in this population? How could you simulate from this
distribution?</p>
<div id="answer" class="section level2">
<h2>Answer</h2>
<div id="the-overall-distribution" class="section level3">
<h3>The overall distribution</h3>
<p>Let <span class="math inline">\(X\)</span> denote the protein
concentration of a randomly chosen individual from the population. Let
<span class="math inline">\(Z\)</span> denote whether the
randomly-chosen individual is normal (<span
class="math inline">\(Z=1\)</span>) or diseased (<span
class="math inline">\(Z=2\)</span>). Here we assume that <span
class="math inline">\(Z\)</span> is not observed; it has been introduced
to help with notation and derivations.</p>
<p>By the law of total probability, we can write the density of <span
class="math inline">\(X\)</span> as <span class="math display">\[
p(x) = \Pr(Z = 1) \, p(x \mid Z = 1) +
\Pr(Z=2) \, p(x \mid Z = 2).
\]</span> In words, this represents <span class="math display">\[
p(x) = \Pr(\text{normal}) \, p(x \mid \text{normal}) +
\Pr(\text{diseased}) \, p(x \mid \text{diseased}).
\]</span></p>
<p>From the information given, we know that <span
class="math inline">\(\Pr(\text{normal})=0.7\)</span> and <span
class="math inline">\(\Pr(\text{diseased}) = 0.3\)</span>. We also know
<span class="math inline">\(p(x \mid \text{normal})\)</span> and <span
class="math inline">\(p(x \mid \text{diseased})\)</span> are each given
by the density of a Gamma distribution. So we can write <span
class="math display">\[
\textstyle
p(x) = 0.7 \, \mathrm{Gamma}(x; 0.5, 2) + 0.3 \, \mathrm{Gamma}(x; 1,
2),
\]</span> where <span class="math inline">\(\mathrm{Gamma}(x; a,
b)\)</span> denotes the density of a Gamma distribution with scale <span
class="math inline">\(a\)</span> and shape <span
class="math inline">\(b\)</span>.</p>
<p>This distribution is an example of a “mixture distribution” (in
particular, it is a “mixture of two Gamma distributions”). Here we plot
the density of this mixture distribution (solid blue), as well as the
densities of the individual distributions that were combined to make the
mixture (the dashed black line for “normal”, the dashed red line for
“diseased”). In mixture terminology, these individual distributions are
called the “component distributions”.</p>
<pre class="r"><code>x &lt;- seq(0,10,length.out = 100)
plot(x,
     0.7 * dgamma(x,scale = 0.5,shape = 2) +
     0.3 * dgamma(x,scale = 1,shape = 2),
     col = &quot;blue&quot;,type = &quot;l&quot;,lwd = 1.5,
     xlab = &quot;protein concentration&quot;,
     ylab = &quot;density&quot;,ylim = c(0,0.8))
lines(x,dgamma(x,scale = 0.5,shape = 2),type = &quot;l&quot;,col = &quot;black&quot;,
      lwd = 1,lty = &quot;dashed&quot;)
lines(x,dgamma(x,scale = 1,shape = 2),type = &quot;l&quot;,col = &quot;red&quot;,lwd = 1,
      lty = &quot;dashed&quot;)</code></pre>
<p><img src="figure/mixture_models_01.Rmd/gammamix-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gammamix-1">
Past versions of gammamix-1.png
</button>
</p>
<div id="fig-gammamix-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/289127545edde17b45e7d89524c55b39e810e607/docs/figure/mixture_models_01.Rmd/gammamix-1.png" target="_blank">2891275</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="simulation" class="section level3">
<h3>Simulation</h3>
<p>One nice thing about mixture models is that they are easy to simulate
from. The trick is to simulate both <span class="math inline">\(X,
Z\)</span> from the joint distribution <span class="math inline">\(p(X,
Z)\)</span>, then simply ignore <span class="math inline">\(Z\)</span>.
This ensures that <span class="math inline">\(X\)</span> comes from its
marginal distribution <span class="math inline">\(p(X)\)</span>.</p>
<p>Simulating from <span class="math inline">\(p(X, Z)\)</span> can be
achieved by a two-stage process: (1) simulate <span
class="math inline">\(Z \sim p(Z)\)</span>, (2) simulate <span
class="math inline">\(X \mid Z \sim p(X \mid Z)\)</span>, both of which
are easy. The following code illustrates this idea by simulating 10,000
samples from the mixture, then plotting a histogram of the samples. As
you can see, the histogram closely matches the mixture density.</p>
<pre class="r"><code>n &lt;- 10000
x &lt;- rep(0,n)
shape &lt;- c(2,2)
scale &lt;- c(0.5,1)
for (i in 1:n) {
  if (runif(1) &lt; 0.7) 
    z &lt;- 1
  else
    z &lt;- 2
  x[i] &lt;- rgamma(1,scale = scale[z],shape = shape[z])
}
hist(x,breaks = 64,xlim = c(0,10),probability = TRUE,main = &quot;&quot;)
xvec &lt;- seq(0,10,length.out = 100)
lines(xvec,
      0.7 * dgamma(xvec,scale = 0.5,shape = 2) +
      0.3 * dgamma(xvec,scale = 1,shape = 2),
      lwd = 1.5,col = &quot;blue&quot;)</code></pre>
<p><img src="figure/mixture_models_01.Rmd/gammamix-sim-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gammamix-sim-1">
Past versions of gammamix-sim-1.png
</button>
</p>
<div id="fig-gammamix-sim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/289127545edde17b45e7d89524c55b39e810e607/docs/figure/mixture_models_01.Rmd/gammamix-sim-1.png" target="_blank">2891275</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note that we set <code>probability = TRUE</code> when creating the
histogram in the above code so that it is scaled like a density (i.e.,
area = 1). This makes it easy to compare with the theoretical
density.</p>
</div>
</div>
</div>
<div id="exercises" class="section level1">
<h1>Exercises</h1>
<p>The following exercises are designed to help you generalize the ideas
in example above to other settings.</p>
<div id="other-mixture-component-distributions" class="section level2">
<h2>1. Other mixture component distributions</h2>
<p>In the above example, there was people sampled from a population that
is 50% male and 50% female, where the males have heights that are
normally distributed with mean 70 in and standard deviation 3 in, and
the females have heights that are normally distributed with mean 64.5 in
and standard deviation 2.5 in. Write the density of the mixture model.
Identify the mixture proportions and the mixture component densities.
Plot the mixture density and the component densities in a plot similar
to the one in the example above.</p>
</div>
<div id="more-than-two-distributions" class="section level2">
<h2>2. More than two distributions</h2>
<p>Similarly, there is nothing that limits us to mixing just two
distributions — you can mix any number of distributions together. This
exercise illustrates this idea.</p>
<p>Suppose that in the protein concentration example above, the
population consists of males and females, who have different rates of
disease, and also different protein distributions. So now there are four
groups: “male, diseased”; “male, normal”; “female, diseased”; and
“female, normal”. Making whatever assumptions you want to make about the
component distributions (say what they are), and about the relative
frequency of each group in the population (again, say what they are),
write out a mixture distribution that could represent this situation.
Plot the components of your assumed mixture and the mixture density.</p>
</div>
<div id="discrete-data" class="section level2">
<h2>3. Discrete data</h2>
<p>The above example involves a mixture of two continuous distributions
(Gamma distributions) and so the mixture distribution is also
continuous. However, mixtures of discrete distributions work in the same
way: you just use probability mass functions instead of probability
density functions.</p>
<p>For example, let <span class="math inline">\(X\)</span> denote the
number of molecules of a particular gene in a cell randomly drawn from
some population of cells. Assume that the population contains three cell
types, in proportions 0.2, 0.4, 0.4. Further suppose that in each cell
type the number of molecules follows a Poisson distribution with mean
parameter <span class="math inline">\(\lambda
= 2, 5, 10\)</span>, respectively. The distribution of <span
class="math inline">\(X\)</span> is therefore a mixture of three Poisson
distributions. Can you write down its probability mass function?</p>
</div>
</div>
<div id="the-general-case" class="section level1">
<h1>The general case</h1>
<p>Putting the above ideas together, we can write the density for a
mixture of <span class="math inline">\(K\)</span> continuous
distributions that have densities <span class="math inline">\(f_1,
\dots, f_K\)</span> in proportions <span class="math inline">\(\pi_1,
\dots, \pi_K\)</span>. Such a mixture would have density <span
class="math display">\[
p(x) = \sum_{k=1}^K \pi_k f_k(x).
\]</span> Exactly the same equation holds for a mixture of <span
class="math inline">\(K\)</span> discrete distibutions, but with <span
class="math inline">\(p(x)\)</span> and <span
class="math inline">\(f_k\)</span> representing probability mass
functions instead of densities.</p>
<div id="some-important-terminology" class="section level2">
<h2>Some important terminology</h2>
<ul>
<li><p>This is called a <em>mixture distribution</em> (or mixture model,
or just mixture) with <span class="math inline">\(K\)</span>
<em>components</em>. (Sometimes it is called a <em>finite mixture</em>
because one can also further generalize the ideas to an uncountably
infinite number of components!)</p></li>
<li><p>The <span class="math inline">\(f_1, \dots, f_K\)</span> are
called the <em>component densities</em> (or <em>component
distributions</em>). So <span class="math inline">\(f_1\)</span> is the
density of component 1, and <span class="math inline">\(f_k\)</span> is
the density of component <span
class="math inline">\(k\)</span>.</p></li>
<li><p>The <span class="math inline">\(\pi_1, \dots, \pi_K\)</span> are
called the <em>mixture proportions</em>. Of course, we must have <span
class="math inline">\(\pi_k \geq 0\)</span> and <span
class="math inline">\(\sum_{k=1}^K \pi_k =1\)</span>.</p></li>
<li><p>The unobserved random variable <span
class="math inline">\(Z\)</span> is sometimes referred to as the
“component of origin” or the “component that gave rise to” the
observation <span class="math inline">\(X\)</span>. If we have <span
class="math inline">\(n\)</span> observations <span
class="math inline">\(X_1,\dots,X_n\)</span> from a mixture model it is
common to use <span class="math inline">\(Z_i\)</span> to denote the
component that gave rise to <span
class="math inline">\(X_i\)</span>.</p></li>
<li><p>Introducing unobserved variables to help with computations or
derivations is a common trick that is used beyond mixture models. This
trick is sometimes called <em>data augmentation</em>. The unobserved
random variables are sometimes called “hidden variables” or “latent
variables”. Representing the mixture model <span class="math display">\[
p(x) = \sum_{k=1}^K
\pi_k f_k(x)
\]</span> by the two-stage process <span class="math display">\[
\begin{aligned}
p(Z=k) &amp;= \pi_k \\
p(x \mid Z = k) &amp;= f_k(x), \quad k = 1, \ldots, K
\end{aligned}
\]</span> is called the <em>latent variable representation</em> of the
mixture model.</p></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
