<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-07" />

<title>Generate data from a simple genetic mixture model</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fiveMinuteStats</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/pcarbo/fiveMinuteStats">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Generate data from a simple genetic mixture
model</h1>
<h4 class="author">Matthew Stephens</h4>
<address class="author_afil">
University of Chicago<br><h4 class="date">January 7, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>fiveMinuteStats/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree3e05e53d387e371ac6d680a62298e0c159ad26d3targetblank3e05e53a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/3e05e53d387e371ac6d680a62298e0c159ad26d3" target="_blank">3e05e53</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcompcarbofiveMinuteStatstree3e05e53d387e371ac6d680a62298e0c159ad26d3targetblank3e05e53a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/pcarbo/fiveMinuteStats/tree/3e05e53d387e371ac6d680a62298e0c159ad26d3" target="_blank">3e05e53</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/r_simplemix.Rmd</code>) and HTML
(<code>docs/r_simplemix.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/3e05e53d387e371ac6d680a62298e0c159ad26d3/analysis/r_simplemix.Rmd" target="_blank">3e05e53</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-07
</td>
<td>
Some updates to the answer templates for r_simplemix.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/4f6df7fefc19bb7c628453390d0e231845429594/analysis/r_simplemix.Rmd" target="_blank">4f6df7f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-07
</td>
<td>
Some updates to the answer template for exercise 2 in r_simplemix.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c217760f2eb0394c6d80b5989691f3cc99133a7c/analysis/r_simplemix.Rmd" target="_blank">c217760</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-07
</td>
<td>
A few small edits to the ‘inference problems’ section of the r_simplemix
vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/f506dc9027343f27a99dd3f04f26c662e2d6422d/analysis/r_simplemix.Rmd" target="_blank">f506dc9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-07
</td>
<td>
Some improvements to the example in the r_simplemix vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/13edb214d3be8be79dd0330d8771c880ddca0e49/analysis/r_simplemix.Rmd" target="_blank">13edb21</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-07
</td>
<td>
Some minor fixes to the r_simplemix vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/6a008f15b708fac71ea6a2ad51b903f669906593/analysis/r_simplemix.Rmd" target="_blank">6a008f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2026-01-06
</td>
<td>
A few small changes to the r_simplemix vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/cb4b7d51ac61d8b79bf18f8d98b9b2273e6ec9b4/analysis/r_simplemix.Rmd" target="_blank">cb4b7d5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Created pdf version of r_simplemix vignette.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/cb4b7d51ac61d8b79bf18f8d98b9b2273e6ec9b4/docs/r_simplemix.html" target="_blank">cb4b7d5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Created pdf version of r_simplemix vignette.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/04994f5dd7af60de1d5764dc2142d8f0b6e5ed23/analysis/r_simplemix.Rmd" target="_blank">04994f5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-12-29
</td>
<td>
Added basic Makefile infrastructure.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/b7835ec7b431f887e22364956d51b78bee496afd/docs/r_simplemix.html" target="_blank">b7835ec</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-07-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/59e192dc1076b7a72b1d579dc69d180623da1fe6/analysis/r_simplemix.Rmd" target="_blank">59e192d</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2021-07-21
</td>
<td>
wflow_publish("r_simplemix.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/e072bd89e6652b9e6d92cd2a08b42337d4a23046/docs/r_simplemix.html" target="_blank">e072bd8</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/2e3ae76ae7b1618dd56335ff9fd4dd795271893c/analysis/r_simplemix.Rmd" target="_blank">2e3ae76</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-27
</td>
<td>
workflowr::wflow_publish("analysis/r_simplemix.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/8dc31ecab87f00b9aebe38cfb4639f854180faaf/docs/r_simplemix.html" target="_blank">8dc31ec</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/087983260a2471584622c6054c34a8b4c519f26b/analysis/r_simplemix.Rmd" target="_blank">0879832</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-26
</td>
<td>
workflowr::wflow_publish("analysis/r_simplemix.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/e2a6440c0f863e98a493435091af916a34525430/analysis/r_simplemix.Rmd" target="_blank">e2a6440</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-26
</td>
<td>
added by hand to avoid figure error
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/e2a6440c0f863e98a493435091af916a34525430/docs/r_simplemix.html" target="_blank">e2a6440</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-26
</td>
<td>
added by hand to avoid figure error
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/pcarbo/fiveMinuteStats/227f78905a240857d7cd6779fc97297ba0a82250/docs/r_simplemix.html" target="_blank">227f789</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/pcarbo/fiveMinuteStats/blob/c95261f16a613d7cc4f8d60874e295543b3519b6/analysis/r_simplemix.Rmd" target="_blank">c95261f</a>
</td>
<td>
Matthew Stephens
</td>
<td>
2020-07-22
</td>
<td>
workflowr::wflow_publish("r_simplemix.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>See <a
href="https://github.com/pcarbo/fiveMinuteStats/blob/master/docs/r_simplemix.pdf">here</a>
for a PDF version of this vignette.</p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette is an “ice breaker” to motivate learning and
statistical inference centered around genetic mixtures.</p>
<p>You don’t need to know what a mixture model is to understand this.
But if you want to know more about mixture models in general you might
read <a
href="https://pcarbo.github.io/fiveMinuteStats/intro_to_mixture_models.html">this
introduction to mixture models</a>.</p>
<div id="simulating-haploid-data-for-a-single-population"
class="section level2">
<h2>Simulating haploid data for a single population</h2>
<p>First we consider simulating genotype data for <span
class="math inline">\(n\)</span> haploid individuals at <span
class="math inline">\(R\)</span> independent biallelic loci (positions
along the genome) sampled from a population.</p>
<p>The term “haploid” means that each individual has only one copy of
their genome. (Most animals are “diploid”, which means they have two
copies of their genome — one inherited from the mother and the other
inherited from the father. However, focussing on haploid individuals
makes the ideas and code easier to follow. Once you understand the
haploid case it is not too hard to extend the ideas to the diploid
case.)</p>
<p>The term “biallelic” means that the loci have two possible alleles
(types), which for convenience we will label 0 and 1.</p>
<p>Under these assumptions, the genotype for each individual is simply a
sequence of zeros and ones. The probability of seeing a 0 vs. a 1 at
each locus is determined by the “allele frequencies” at each locus,
which we will specify by a vector <span
class="math inline">\(p\)</span>. Specifically, <span
class="math inline">\(p_r\)</span> specifies the frequency of the 1
allele at locus <span class="math inline">\(r\)</span>.</p>
<p>The following code simulates from this model.</p>
<pre class="r"><code>#&#39; @param n The number of samples.
#&#39; @param p A vector of allele frequencies for R loci.
#&#39; @return An n x R matrix of haploid genotypes.
r_haploid_genotypes &lt;- function (n, p) {
  R &lt;- length(p)
  x &lt;- matrix(0,n,R)
  for (i in 1:n)
    x[i,] &lt;- rbinom(R,rep(1,R),p)
  return(x)
}</code></pre>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<p>To illustrate this function, we simulate a small example dataset
containing 20 haploid individuals at 9 loci. The frequencies of the 1
allele at the loci are increasing from 0.1 at the first locus to 0.9 at
the ninth locus. (The pattern is not supposed to be realistic, it is
just to help illustrate the idea.)</p>
<pre class="r"><code>set.seed(123)
p &lt;- seq(0.1,0.9,length.out = 9)
x &lt;- r_haploid_genotypes(20,p)
p
# [1] 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9
x 
#       [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9]
#  [1,]    0    0    0    1    1    1    1    0    1
#  [2,]    0    1    0    1    1    1    0    1    1
#  [3,]    0    1    1    1    1    0    1    1    1
#  [4,]    0    0    0    1    1    0    0    1    1
#  [5,]    0    0    0    0    0    1    1    1    1
#  [6,]    0    0    0    0    1    1    1    1    1
#  [7,]    0    0    0    1    1    1    1    1    1
#  [8,]    0    1    0    1    1    0    1    1    1
#  [9,]    0    0    0    0    0    0    1    1    1
# [10,]    0    0    1    0    0    0    0    0    1
# [11,]    0    0    0    1    0    1    0    1    1
# [12,]    0    0    0    0    1    1    0    0    1
# [13,]    0    0    1    0    0    0    0    1    1
# [14,]    1    0    0    1    0    1    1    1    0
# [15,]    0    0    0    1    1    0    1    1    1
# [16,]    0    1    1    1    0    1    1    1    1
# [17,]    0    0    0    0    0    0    0    1    1
# [18,]    0    0    0    0    0    1    1    1    1
# [19,]    0    0    0    0    1    1    1    1    1
# [20,]    0    1    1    1    1    1    1    0    1</code></pre>
<p>As you can see, the 1 allele is rarer at the earlier loci, whereas
the 0 allele is rarer at the later loci.</p>
</div>
</div>
<div id="simulating-haploid-data-from-a-mixture-of-two-populations"
class="section level1">
<h1>Simulating haploid data from a mixture of two populations</h1>
<p>Now suppose we sample from a group of individuals formed by mixing
together the individuals from two different populations. This is an
example of a “mixture model”.</p>
<p>For simplicity, we will assume the two different populations are
mixed in equal proportions. That is, the “mixture proportions” are 0.5
and 0.5.</p>
<p>The following function generates data from such a model. The allele
frequencies for the two populations must be specified in a matrix, <span
class="math inline">\({\bf P}\)</span>, whose first row contains the
allele frequencies for population 1 and second row is the allele
frequencies for population 2. (So element <span
class="math inline">\(p_{kr}\)</span> is the frequency of the 1 alleles
in population <span class="math inline">\(k\)</span> at locus <span
class="math inline">\(r\)</span>.) For each individual <span
class="math inline">\(i\)</span>, it randomly samples the population
<span class="math inline">\(z_i = k\)</span> to be 1 or 2, and then uses
the <code>r_haploid_genotypes</code> function to generate the genotypes
from that population.</p>
<pre class="r"><code>#&#39; @param n The number of samples.
#&#39; @param P A 2 x R matrix of allele frequencies.
#&#39; @return A list containing a matrix of genotypes (x) and
#&#39;   the populations of origin (z).
r_simplemix &lt;- function (n, P) {
  R &lt;- ncol(P)
  z &lt;- rep(0,n)
  x &lt;- matrix(0,n,R)
  for (i in 1:n) {
    z[i]  &lt;- sample(2,1,prob = c(0.5,0.5))
    k     &lt;- z[i]
    x[i,] &lt;- r_haploid_genotypes(1,P[k,])
  }
  return(list(x = x,z = z))
}</code></pre>
<div id="example-1" class="section level2">
<h2>Example</h2>
<p>In this example, we sample 20 individuals: in one population, the
frequencies are as above, whereas in the second population they are
reversed (<span class="math inline">\(1 - p\)</span>). Again, this is
not at all realistic, but intended to illustrate the ideas.</p>
<pre class="r"><code>set.seed(123)
P &lt;- rbind(p,1 - p,deparse.level = 0)
print(P)
#      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9]
# [1,]  0.1  0.2  0.3  0.4  0.5  0.6  0.7  0.8  0.9
# [2,]  0.9  0.8  0.7  0.6  0.5  0.4  0.3  0.2  0.1
sim &lt;- r_simplemix(20,P)</code></pre>
<p>Here are the results of the simulation:</p>
<pre class="r"><code>colnames(sim$x) &lt;- paste0(&quot;r&quot;,1:9)
cbind(z = sim$z,sim$x)
#       z r1 r2 r3 r4 r5 r6 r7 r8 r9
#  [1,] 2  1  1  0  0  0  0  1  0  0
#  [2,] 1  0  0  0  0  1  1  1  1  0
#  [3,] 1  0  0  1  1  1  1  1  1  1
#  [4,] 1  1  0  1  0  0  0  1  1  1
#  [5,] 2  1  1  1  1  0  0  0  0  0
#  [6,] 2  1  1  1  1  0  0  1  1  0
#  [7,] 1  0  0  0  1  0  0  0  1  1
#  [8,] 1  0  0  0  0  0  1  1  1  1
#  [9,] 2  1  1  0  1  0  1  1  1  0
# [10,] 2  1  1  1  1  0  1  0  0  0
# [11,] 1  0  0  1  0  1  0  1  1  1
# [12,] 1  0  0  1  1  0  1  0  1  1
# [13,] 1  0  0  0  0  1  1  1  1  1
# [14,] 1  0  0  1  0  1  0  0  0  1
# [15,] 2  1  1  1  0  0  0  0  0  0
# [16,] 1  0  0  0  0  0  1  1  1  1
# [17,] 1  0  0  0  0  1  0  1  1  1
# [18,] 1  0  1  1  1  1  1  1  0  1
# [19,] 1  0  0  0  0  0  1  1  0  0
# [20,] 2  1  0  1  0  0  0  0  0  0</code></pre>
<p>If you look carefully, you should see that individuals from
population 1 tend to have more 1 alleles in the later loci, whereas
individuals from population 2 tend to have more 1 alleles in the earlier
loci. This is because of the way that the allele frequencies were set up
in the two populations. Of course, in real data the differences between
different populations will not usually show patterns like this! I chose
the patterns so you can see them by eye.</p>
</div>
</div>
<div id="inference-problems" class="section level1">
<h1>Inference problems</h1>
<p>These types of data can help motivate a number of statistical
inference problems, including:</p>
<ol style="list-style-type: decimal">
<li><p>Given the allele frequencies <span class="math inline">\({\bf
P}\)</span> and the genotypes <span class="math inline">\({\bf
X}\)</span>, how might you infer the populations of origin <span
class="math inline">\(z\)</span>? In the genetic literature, this is
sometimes called the “assignment problem”; more generally, it is an
example of a “classification problem”.</p></li>
<li><p>Given the populations of origin <span
class="math inline">\(z\)</span> and the genotypes <span
class="math inline">\({\bf X}\)</span>, how might you infer the
population allele frequencies <span class="math inline">\({\bf
P}\)</span>?</p></li>
<li><p>Given just the genotypes <span class="math inline">\({\bf
X}\)</span> how might you infer <em>both</em> <span
class="math inline">\(z\)</span> and <span class="math inline">\({\bf
P}\)</span>? This is an example of a “clustering problem”.</p></li>
</ol>
</div>
<div id="exercises" class="section level1">
<h1>Exercises</h1>
<p>Here are some things you might like to try:</p>
<ol style="list-style-type: decimal">
<li><p>Modify the <code>r_simplemix</code> function to allow the mixture
proportions to be specified as an input argument, rather than fixed at
<span class="math inline">\((0.5, 0.5)\)</span>. You could do this by
adding a parameter “w” to the function that specifies the proportions to
use (“w” is short for “weights”).</p></li>
<li><p>Write a function <code>posterior_prob_assignment(x,P,w)</code> to
compute the posterior probability that each individual came from each
population, given the genotypes “x”, the allele frequencies “P” and the
mixture proportions “w”. Apply your function to the data you
simulated.</p></li>
<li><p>Write a function
<code>posterior_param_allele_frequencies(x,z,a)</code> to compute the
parameters of the Beta posterior in each population at each locus. Here,
“a” is a vector of length 2 giving the parameters of the Beta prior for
<span class="math inline">\(p_{kr}\)</span>; that is, the prior is <span
class="math inline">\(p_{kr} \sim
\mathrm{Beta}(a_1, a_2)\)</span>. Because the Beta distribution has two
parameters, there will be 2 parameters for each locus and for each
population. So the output of your function should be two <span
class="math inline">\(2 \times R\)</span> matrices, or a <span
class="math inline">\(2 \times 2 \times R\)</span> array.</p></li>
</ol>
</div>
<div id="answer-templates" class="section level1">
<h1>Answer templates</h1>
<p>Here are some answer templates for Exercises 2 and 3. You will need
to fill in the “add your code here” comments. (<em>Note:</em> some of
these code chunks have the option set <code>eval = FALSE</code> so that
this document compiles. You will have to remove this option if you want
these chunks to run on compilation.)</p>
<div id="exercise-2" class="section level2">
<h2>Exercise 2</h2>
<p>Function to compute the likelihood:</p>
<pre class="r"><code>#&#39; @param p A vector of allele frequencies of length R.
#&#39; @param x A vector of genotypes of length R.
#&#39; @return The likelihood for the allele frequences (p) given the
#&#39;   genotype data (x).
likelihood &lt;- function (p, x)
  prod(p^x*(1-p)^(1-x))</code></pre>
<p>Function to “normalize” a vector:</p>
<pre class="r"><code>#&#39; @param x A vector.
#&#39; @return The normalized vector in which its elements sum to 1.
normalize &lt;- function (x)
  x/sum(x)</code></pre>
<p>Function to compute the posterior assignment probabilities:</p>
<pre class="r"><code>#&#39; @param x An n x R matrix of genotypes.
#&#39; @param P A 2 x R matrix of allele frequencies.
#&#39; @param w A vector of mixture proportions of length 2.
#&#39; @return An n x 2 matrix of posterior assignment probabilities.
posterior_prob_assignment &lt;- function (x, P, w) {
  K &lt;- 2
  n &lt;- nrow(x)
  post &lt;- matrix(0,n,K)
  lik &lt;- rep(0,K)
  for (i in 1:n) {
    for (k in 1:K) {
      # Add your code here.
    }
    # Add your code here.
  }
  return(post)
}</code></pre>
<p>Once you have implemented this function, try running it on an example
data set, and compare the posterior assignment probabilities to the true
labels:</p>
<pre class="r"><code>set.seed(123)
sim &lt;- r_simplemix(20,P)
posterior &lt;- posterior_prob_assignment(sim$x,P,w = c(0.5,0.5))
plot(sim$z,posterior[,2],xlab = &quot;true population&quot;,
     ylab = &quot;posterior prob for population 2&quot;)</code></pre>
</div>
<div id="exercise-3" class="section level2">
<h2>Exercise 3</h2>
<p>Function to compute the posterior distribution of the allele
frequencies for each population and for each locus.</p>
<pre class="r"><code>#&#39; @param x An n by R matrix of genotypes.
#&#39; @param z An vector of population assignments of length n.
#&#39; @param a A vector of length 2 specifying the Beta prior.
#&#39; @return A 2 x 2 x R array containing the posterior parameters for the
#&#39;    the allele frequencies given the genotypes and population labels.
posterior_param_allele_frequencies &lt;- function (x, z, a) {
 K &lt;- 2
 R &lt;- ncol(x)
 post_param &lt;- array(0,c(2,K,R))
 for (k in 1:K) {
   for (r in 1:R) {
     # Add your code here.
   }
 }
 return(post_param)
}</code></pre>
<p>Once you have implemented this function, try running it on an example
data set</p>
<pre class="r"><code>set.seed(123)
sim &lt;- r_simplemix(20,P)
posterior_param &lt;- posterior_param_allele_frequencies(sim$x,sim$z,a = c(1,1))
posterior_param[,1,] # Posterior for population 1.
posterior_param[,2,] # Posterior for population 2.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.7.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# loaded via a namespace (and not attached):
#  [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
#  [5] xfun_0.52         stringi_1.8.7     promises_1.3.3    jsonlite_2.0.0   
#  [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.33.0     
# [13] htmltools_0.5.8.1 httpuv_1.6.14     sass_0.4.10       rmarkdown_2.29   
# [17] evaluate_1.0.4    jquerylib_0.1.4   tibble_3.3.0      fastmap_1.2.0    
# [21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
# [25] compiler_4.3.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
# [29] later_1.4.2       digest_0.6.37     R6_2.6.1          pillar_1.11.0    
# [33] magrittr_2.0.3    bslib_0.9.0       tools_4.3.3       cachem_1.1.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
